---
title: Efferocytosis Remodels the Arachidonic Acid Program to Distinguish the Newborn Regenerative Response
author: "Connor W Lantz"
output: html_document
---
# Summary 

In response to organ injury, macrophages found in the heart enhance scarring, yet in early life support tissue regeneration. To elucidate underlying mechanisms, we compared the macrophage response in newborns to adults after injury. Sequencing revealed a neonatal-specific accumulation of tissue-resident macrophages that were selectively polarized for apoptotic cell recognition (or efferocytosis). Genetic ablation of apoptotic cell receptors transformed the neonatal phenotype to one that resembled adult macrophages along with tissue scarring. This could be explained by remodeling of macrophage gene expression required for arachidonic acid metabolism and biosynthesis of the eicosanoid thromboxane, the latter which unexpectedly activated parenchymal cell proliferation. Thromboxane receptors were discovered on neighboring cardiomyocytes and were required for a cellular metabolic shift which was coupled to mitogenic signaling. These data are consistent with a fundamental age-defined macrophage response in which lipid mitogens produced during efferocytosis support organ regeneration.

# Loading R Packages and Single-Cell Data
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE)
```

### Set Working Directory and Load Packages
```{r}
setwd("/projects/p31275/Lantz/R/Lantz_etal_Manuscript")
```

### Load Packages for Analyses
```{r packages, message=FALSE, warning=FALSE, include=FALSE}
library(pkgbuild)
library(ggplot2)
library(dplyr)
library(Matrix)
library(sctransform)
library(cowplot)
library(Seurat)
library(umap)
library(RCurl)
library(scales)
library(tidyverse)
library(RColorBrewer)
library(SingleR)
library(celldex)
library(scran)
library(glmGamPoi)
library(scater)
library(gridExtra)
library(dyno)
library(monocle3)
library(SeuratWrappers)
library(EnhancedVolcano)
library(ragg)
library(ComplexHeatmap)
library(CellChat)
options(future.globals.maxSize= 15001289600)
```

### Load Output Data from CellRanger Pipeline
For each condition's dataset, a Seurat Object is created by reading the corresponding 10X dataset. Low quality cells with less than 100 features are excluded from each Seurat object.
```{r Load CellRanger Matrices}
for (file in c("WT_P1_Sh", "WT_P1_MI_1", "WT_P1_MI_2", "WT_Ad_Sh", "WT_Ad_MI", "MER_Sh", "MER_MI")){
  seurat_data <- Read10X(data.dir = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/DATA/", file))
  seurat_obj <- CreateSeuratObject(counts = seurat_data, min.features = 100, project = file)
  assign(file, seurat_obj)
}

#Merge Seurat Object for QC
merged_seurat <- merge(x = WT_P1_Sh, y = c(WT_P1_MI_1, WT_P1_MI_2, WT_Ad_Sh, WT_Ad_MI, MER_Sh, MER_MI), add.cell.ids = c("WT_P1_Sh", "WT_P1_MI_1", "WT_P1_MI_2", "WT_Ad_Sh", "WT_Ad_MI", "MER_Sh", "MER_MI"))

merged_seurat
```

# Quality Control
### Calculating % Mitochondrial and % Ribosomal Reads
The percent of mitochondrial and ribosomal reads can be informative for the "health" of the cells. Typically high mitochondrial and ribosomal content may indicate dying cells which should be excluded from downstream applications.
```{r Calculate QC Metrics}
#Log Normalization of Merged Seurat
merged_seurat$log10GenesPerUMI <- log10(merged_seurat$nFeature_RNA) / log10(merged_seurat$nCount_RNA)

#Mitochondrial Feature Expression Ratio
merged_seurat$mitoRatio <- PercentageFeatureSet(merged_seurat, pattern = "^mt-")
merged_seurat$mitoRatio <- merged_seurat@meta.data$mitoRatio / 100

#Ribosomal Feature Expression Ratio
merged_seurat$riboRatio <- PercentageFeatureSet(merged_seurat, pattern = "^Rp[ls]")
merged_seurat$riboRatio <- merged_seurat@meta.data$riboRatio / 100
```

### Create Metadata Slot
The metadata slot is a handy place to store information about the dataset. For example, we stored the % Mitochondrial and % Ribosomal Reads in the metadata slot above and now we are adding our sample identification so that we can easily parse cells based on sample identity. This will allow us to easily compare between our samples in downstream analyses.
``` {r Create Metadata slot}
#Create Metadata Slot in Merged Seurat
metadata <- merged_seurat@meta.data
metadata$cells <- rownames(metadata)
metadata <- metadata %>%
  dplyr::rename(seq_folder = orig.ident,
                nUMI = nCount_RNA,
                nGene = nFeature_RNA)

#Add Identification Metadata for Each Condition
metadata$sample <- NA
metadata$sample[which(str_detect(metadata$cells, "^WT_P1_Sh"))] <- "WT_P1_Sh"
metadata$sample[which(str_detect(metadata$cells, "^WT_P1_MI"))] <- "WT_P1_MI"
metadata$sample[which(str_detect(metadata$cells, "^WT_Ad_Sh"))] <- "WT_Ad_Sh"
metadata$sample[which(str_detect(metadata$cells, "^WT_Ad_MI"))] <- "WT_Ad_MI"
metadata$sample[which(str_detect(metadata$cells, "^MER_Sh"))] <- "MER_Sh"
metadata$sample[which(str_detect(metadata$cells, "^MER_MI"))] <- "MER_MI"


merged_seurat@meta.data <- metadata

head(merged_seurat@meta.data, 5)
```
The output reveals the metadata matrix. The first column indicates the cellular barcode from 10x Genomics pipeline. The other columns consist of our sample ID along with the QC metrics calculated previously.

### Displaying QC metrics
Different quality control metrics such as the number of UMIs, genes, mitochondrial, and ribosomal counts are analyzed to determine the quality of the data. The results from these different quality control metrics will be used to filter out low-quality cells.

```{r Quality Control Metrics, echo=TRUE, warning=FALSE}
levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI", "MER_Sh", "MER_MI")
merged_seurat@meta.data$sample <- factor(x = merged_seurat@meta.data$sample, levels = levels)
metadata$sample <- factor(x = metadata$sample, levels = levels)

#nCells
p1 <- metadata %>% 
  ggplot(aes(x=factor(x = merged_seurat$sample, levels = levels), fill=sample)) + 
  geom_bar() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("nCells") +
  NoLegend()

#nUMIs
p2 <- metadata %>% 
  ggplot(aes(color = sample, x = nUMI, fill = sample)) + 
  geom_density(alpha = 0.2) + 
  scale_x_log10() + 
  theme_classic() +
  ylab("Cell density") +
  geom_vline(xintercept = 500) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("nUMIS")

#nGenes 
p3 <- metadata %>% 
  ggplot(aes(x=factor(x = merged_seurat$sample, levels = levels), y=log10(nGene), fill=sample)) + 
  geom_boxplot() + 
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("nGenes") +
  NoLegend()

#mito counts ratio
p4 <- metadata %>% 
  ggplot(aes(color=sample, x=mitoRatio, fill=sample)) + 
  geom_density(alpha = 0.1) + 
  scale_x_log10() + 
  theme_classic() +
  geom_vline(xintercept = 0.3) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("Mito Ratio") +
  NoLegend()

#ribo counts ratio
p5 <- metadata %>% 
  ggplot(aes(color=sample, x=riboRatio, fill=sample)) + 
  geom_density(alpha = 0.2) + 
  scale_x_log10() + 
  theme_classic() +
  geom_vline(xintercept = 0.5) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("Ribo Ratio") +
  NoLegend()

#novelty gene
p6 <- metadata %>%
  ggplot(aes(x=log10GenesPerUMI, color = sample, fill=sample)) +
  geom_density(alpha = 0.2) +
  theme_classic() +
  geom_vline(xintercept = 0.8) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("Unique genes per UMI") +
  NoLegend()

plot1 <- grid.arrange(p1,p2,p3,p4,p5,p6, ncol = 2)
plot1
```

```{r eval=FALSE, include=FALSE}
ggsave(filename = "QCmetrics.tiff", plot = plot1, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Quality_Control", device = "tiff", dpi = 500, width = 7, height = 7, units = "in", compression = "lzw")
```

We can also compare nGene versus nUMI to determine cells that have low sequencing qualifications. For example, cells with fewer genes or UMIs would not be informative in downstream analyses and thus will be filtered out. We can plot this and color each cell by the mitochondrial content. Not surprisingly, cells with low nUMIs and low nGenes tend to also be high in %mito content. All together these QC metrics indicate these cells should be removed from our analyses.
```{r Genes versus UMI Correlations, echo=TRUE, message=FALSE, warning=FALSE}
#Correlation of genes v. UMI
plot2 <- metadata %>% 
  ggplot(aes(x=nUMI, y=nGene, color=mitoRatio)) + 
  geom_point() + 
  scale_colour_gradient(low = "deepskyblue3", high = "firebrick4") +
  stat_smooth(method=lm) +
  scale_x_log10() + 
  scale_y_log10() + 
  theme_classic() +
  geom_vline(xintercept = 500) +
  geom_hline(yintercept = 250) +
  facet_wrap(~sample)

plot2
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "nGene_v_nUMI.tiff", plot = plot2, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Quality_Control", device = "tiff", dpi = 500, width = 7, height = 5, units = "in", compression = "lzw")
```

### Filtering Cells Based on QC
Based on the QC results above, we can now filter out "low-quality" cells from the dataset. We decided to remove cells with less than 500 UMIs, less than 300 unique genes, and mitochondrial content higher than 50%.
```{r Filter, message=FALSE}
#Subset high-quality cells on Determined QC metrics from above
filtered_seurat <- subset(x = merged_seurat, 
                          subset= (nUMI >= 500) & 
                          (nGene >= 250) & 
                          (log10GenesPerUMI > 0.80) & 
                          (mitoRatio < 0.30))

#Keep genes expressed in 10 or more cells
counts <- GetAssayData(object = filtered_seurat, slot = "counts")
nonzero <- counts > 0
keep_genes <- Matrix::rowSums(nonzero) >= 3
filtered_counts <- counts[keep_genes, ]

#Create Seurat Object after Filtering Low Quality Cells
filtered_seurat <- CreateSeuratObject(filtered_counts, meta.data = filtered_seurat@meta.data)
filtered_seurat
```
A total of 1,853 cells were removed. Our filtered dataset therefore has 41,265 cells with 22,165 unique features.  

### Revisiting QC Metrics with Filterd Data
```{r Quality Control Metrics, echo=TRUE, warning=FALSE}
metadata <- filtered_seurat@meta.data
#nCells
p1 <- metadata %>% 
  ggplot(aes(x=sample, fill=sample)) + 
  geom_bar() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("nCells") +
  NoLegend()

#nUMIs
p2 <- metadata %>% 
  ggplot(aes(color=sample, x=nUMI, fill= sample)) + 
  geom_density(alpha = 0.2) + 
  scale_x_log10() + 
  theme_classic() +
  ylab("Cell density") +
  geom_vline(xintercept = 500) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("nUMIS")

#nGenes 
p3 <- metadata %>% 
  ggplot(aes(x=sample, y=log10(nGene), fill=sample)) + 
  geom_boxplot() + 
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("nGenes") +
  NoLegend()

#mito counts ratio
p4 <- metadata %>% 
  ggplot(aes(color=sample, x=mitoRatio, fill=sample)) + 
  geom_density(alpha = 0.1) + 
  scale_x_log10() + 
  theme_classic() +
  geom_vline(xintercept = 0.6) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("Mito Ratio") +
  NoLegend()

#ribo counts ratio
p5 <- metadata %>% 
  ggplot(aes(color=sample, x=riboRatio, fill=sample)) + 
  geom_density(alpha = 0.2) + 
  scale_x_log10() + 
  theme_classic() +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("Ribo Ratio") +
  NoLegend()

#novelty gene
p6 <- metadata %>%
  ggplot(aes(x=log10GenesPerUMI, color = sample, fill=sample)) +
  geom_density(alpha = 0.2) +
  theme_classic() +
  geom_vline(xintercept = 0.8) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("Unique genes per UMI") +
  NoLegend()

plot1 <- grid.arrange(p1,p2,p3,p4,p5,p6, ncol = 2)
plot1

plot2 <- metadata %>% 
  ggplot(aes(x=nUMI, y=nGene, color=mitoRatio)) + 
  geom_point() + 
  scale_colour_gradient(low = "deepskyblue3", high = "firebrick4") +
  stat_smooth(method=lm) +
  scale_x_log10() + 
  scale_y_log10() + 
  theme_classic() +
  geom_vline(xintercept = 500) +
  geom_hline(yintercept = 250) +
  facet_wrap(~sample)
plot2

table(metadata$sample)
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "QCmetrics.tiff_filtered.tiff", plot = plot1, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Quality_Control", device = "tiff", dpi = 500, width = 7, height = 7, units = "in", compression = "lzw")

ggsave(filename = "nGene_v_nUMI_filtered.tiff", plot = plot2, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Quality_Control", device = "tiff", dpi = 500, width = 7, height = 5, units = "in", compression = "lzw")
```



# Integration of Neonatal and Adult Datasets After MI
## Cell Cycle Scoring
### Prepare Seurat for Cell Cycle Scoring
```{r}
# A list of cell cycle markers, from Tirosh et al, 2015, is loaded with Seurat. We can segregate this list into markers of G2/M phase and markers of S phase
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes

# Create our Seurat object and complete the initialization steps
filtered_seurat <- NormalizeData(filtered_seurat)
filtered_seurat <- FindVariableFeatures(filtered_seurat, selection.method = "vst")
filtered_seurat <- ScaleData(filtered_seurat, features = rownames(filtered_seurat))
filtered_seurat <- RunPCA(filtered_seurat, features = VariableFeatures(filtered_seurat), ndims.print = 1:10, nfeatures.print = 10)

#Assign Cell Cycle Scores
filtered_seurat <- CellCycleScoring(filtered_seurat,
                                    g2m.features = g2m.genes,
                                    s.features = s.genes,
                                    set.ident = T)

#Separate cycling and non-cycling cells
filtered_seurat$CC.Difference <- filtered_seurat$S.Score - filtered_seurat$G2M.Score
```

### Save Filtered Seurat Before Normalization and Integration
```{r}
saveRDS(filtered_seurat, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/filtered_seurat.rds")
```

## scTransform Normalization
```{r}
#Read in filtered Seurat RDS object
filtered_seurat <- readRDS("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/filtered_seurat.rds")

#Split merged filtered Seurat into individual Seurat Objects
split_seurat <- SplitObject(filtered_seurat, split.by = "sample")

#scTransform data without regressing out mitochondrial genes
for (i in 1:length(split_seurat)) {
  split_seurat[[i]] <- SCTransform(split_seurat[[i]], vst.flavor = "v2", vars.to.regress = c("CC.Difference", "mitoRatio"), method = "glmGamPoi", verbose = F)
}
```

## Integration of Normalized Datasets
### Perform Integration for Seurat Objects normalized and scaled by sctransform
```{r}
#Prepare and Integrate SCTransformed data into an integrated Seurat Object
integ_features <- SelectIntegrationFeatures(object.list = split_seurat, nfeatures = 4000)
split_seurat <- PrepSCTIntegration(object.list = split_seurat, anchor.features = integ_features, verbose = F)
integ_anchors <- FindIntegrationAnchors(object.list = split_seurat, normalization.method = "SCT", anchor.features = integ_features, verbose = F)
Integrated_All_Cells <- IntegrateData(anchorset = integ_anchors, normalization.method = "SCT", verbose = F)
```

### Save RDS for Integrated Seurat Object
```{r}
saveRDS(Integrated_All_Cells, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Integrated_All_Cells.rds")
Integrated_All_Cells <- readRDS("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Integrated_All_Cells.rds")
```

### Perform an Integrated Analysis to confirm no batch effect
```{r}
# Run Principal Component Analysis on Integrated Data
DefaultAssay(Integrated_All_Cells) <- "integrated"
Integrated_All_Cells <- RunPCA(Integrated_All_Cells, npcs = 100, verbose = FALSE)
ElbowPlot(object = Integrated_All_Cells, ndims = 100)

# Reduce Dimensions with UMAP
Integrated_All_Cells <- RunUMAP(Integrated_All_Cells, reduction = "pca", dims = 1:60, verbose = FALSE)
```

#### Plot resulting UMAP split by sample to confirm batch effects are mitigated
```{r}
levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI", "MER_Sh", "MER_MI")
Integrated_All_Cells$sample <- factor(x = Integrated_All_Cells$sample, levels = levels)

Idents(Integrated_All_Cells) <- "sample"
Corrected_Batch_Dimplot <- DimPlot(Integrated_All_Cells, split.by = "sample", cols = c("lightskyblue2", "dodgerblue4", "gray75", "gray25", "darksalmon", "firebrick4"))
Corrected_Batch_Dimplot
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Corrected_Batch_Dimplot.tiff", plot = Corrected_Batch_Dimplot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Quality_Control", device = "tiff", dpi = 500, width = 15, height = 4, units = "in", compression = "lzw")
```

#### Re-visit QC metrics in UMAP projection
```{r}
metrics <- c("nUMI", "nGene", "S.Score", "G2M.Score", "mitoRatio", "riboRatio")
FeaturePlot <- FeaturePlot(Integrated_All_Cells, reduction = "umap", features = metrics, pt.size = 0.4, min.cutoff = "q10", order = T, cols = c("gray89", "firebrick4"), combine = F)

for(i in 1:length(FeaturePlot)) {
  FeaturePlot[[i]] <- FeaturePlot[[i]] + theme(axis.title.y = element_blank(), axis.title.x = element_blank(), axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks.y = element_blank())
}
FeaturePlot <- cowplot::plot_grid(plotlist = FeaturePlot, ncol = 3)
FeaturePlot
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "FeaturePlot_QC_metrics.tiff", plot = FeaturePlot, path = "/projects/p31275/Lantz/R/Lantz_Neonate_Adult_MI/FIGURES/Quality_Control", device = "tiff", dpi = 300, width = 15, height = 8, units = "in", compression = "lzw")
```

# Cluster Identification and Macrophage Heterogeneity at Steady State
## SingleR Annotiations
To unbiasedly identify resident peritoneal cells present in the dataset, SingleR (v1.0.5) was employed. Briefly, SingleR infers the origin of each individual cell by referencing transcriptomic datasets of pure cell types. We utilized the ImmGen database, which contains normalized expression values for immune cells from 830 murine microarrays to ID our peritoneal cell types. 

### Load Reference Mouse Dataset
```{r message=FALSE}
#Load SingleR Package
immgen.se <- ImmGenData()
```

### Run SingleR on Dataset
```{r Single R, message=FALSE, warning=FALSE}
#Label Cells with Main Categories in ImmGen Database
counts <- GetAssayData(Integrated_All_Cells)
pred.Integrated_All_Cells <- SingleR(test = counts, ref = immgen.se, labels = immgen.se$label.main, de.method="wilcox")
Main.labels <- table(pred.Integrated_All_Cells$labels)
Main.labels
```

### Plot Annotation Diagnostics
```{r}
#Create Heatmap based on SingleR classifications
SingleR_Heatmap <- SingleR_Heatmap <- plotScoreHeatmap(pred.Integrated_All_Cells)
SingleR_Heatmap

#Plot Delta Distribution (Low deltas indicate the assignment is uncertain)
SingleR_PlotDistribution_Deltas <- plotDeltaDistribution(pred.Integrated_All_Cells, ncol = 3)
SingleR_PlotDistribution_Deltas
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "SingleR_Heatmap.tiff", plot = SingleR_Heatmap, path = "/projects/p31275/Lantz/R/Lantz_Neonate_Adult_MI/FIGURES/SingleR", device = "tiff", dpi = 300, width = 10, height = 6, units = "in", compression = "lzw")

ggsave(filename = "SingleR_PlotDistribution_Deltas.tiff", plot = SingleR_PlotDistribution_Deltas, path = "/projects/p31275/Lantz/R/Lantz_Neonate_Adult_MI/FIGURES/SingleR", device = "tiff", dpi = 300, width = 8, height = 12, units = "in", compression = "lzw")
```

### Determine which cells were pruned
```{r message=FALSE, warning=FALSE}
to.remove <- is.na(pred.Integrated_All_Cells$pruned.labels)
table(Label=pred.Integrated_All_Cells$labels, Removed=to.remove)
```

### Check if Marker Expression Correlates with Annotations
```{r}
Integrated_All_Cells$SingleR.labels <- pred.Integrated_All_Cells$labels
Integrated_All_Cells.sce <- as.SingleCellExperiment(Integrated_All_Cells)

all.markers <- metadata(pred.Integrated_All_Cells)$de.genes
empirical.markers <- findMarkers(Integrated_All_Cells.sce, Integrated_All_Cells.sce$SingleR.labels, direction="up")


collected <- list()
for (lab in unique(pred.Integrated_All_Cells$labels)) {
    if(lab == "B cells, pro") next
    lab.markers <- unique(unlist(all.markers[[lab]]))
    m <- match(lab.markers, rownames(empirical.markers[[lab]]))
    m <- lab.markers[rank(m) <= 20]
    collected[[lab]] <- plotHeatmap(Integrated_All_Cells.sce, silent = T, main = lab, order_columns_by="SingleR.labels", features=m, legend = F, annotation_legend = F)[[4]]
}

SingleR_Marker_gene_heatmap <- do.call(gridExtra::grid.arrange, collected)
SingleR_Marker_gene_heatmap
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "SingleR_Marker_gene_heatmap.tiff", plot = SingleR_Marker_gene_heatmap, path = "/projects/p31275/Lantz/R/Lantz_Neonate_Adult_MI/FIGURES/SingleR", device = "tiff", dpi = 300, width = 20, height = 15, units = "in", compression = "lzw")
```

## Seurat Clustering 
Clusters were identified using graph-based clustering approach implemented by the FindCluster function in Seurat at various resolutions to determine transcriptionally distinct populations. 
```{r}
DefaultAssay(Integrated_All_Cells) <- "integrated"
Integrated_All_Cells <- FindNeighbors(object = Integrated_All_Cells, dims = 1:60, verbose = F)
Integrated_All_Cells <- FindClusters(object = Integrated_All_Cells, resolution = c(0.2, 0.4, 0.6, 0.8, 1.0, 1.4), verbose = F)

#Resolution 0.4
Idents(object = Integrated_All_Cells) <- "integrated_snn_res.0.4"
Integrated_All_Cells$Res0.4 <- Idents(object = Integrated_All_Cells)
p7 <- DimPlot(Integrated_All_Cells, reduction = "umap", label = T, label.size = 4) +NoLegend()


#Resolution 0.6
Idents(object = Integrated_All_Cells) <- "integrated_snn_res.0.6"
Integrated_All_Cells$Res0.6 <- Idents(object = Integrated_All_Cells)
p8 <- DimPlot(Integrated_All_Cells, reduction = "umap", label = T, label.size = 4)  +NoLegend()


#Resolution 0.8
Idents(object = Integrated_All_Cells) <- "integrated_snn_res.0.8"
Integrated_All_Cells$Res0.8 <- Idents(object = Integrated_All_Cells)
p9 <- DimPlot(Integrated_All_Cells, reduction = "umap", label = T, label.size = 4) + NoLegend()

#Resolution 1.0
Idents(object = Integrated_All_Cells) <- "integrated_snn_res.1"
Integrated_All_Cells$Res1.0 <- Idents(object = Integrated_All_Cells)
p10 <- DimPlot(Integrated_All_Cells, reduction = "umap", label = T, label.size = 4) + NoLegend()

Clustering_UMAPs <- grid.arrange(p7,p8,p9,p10, ncol = 2)
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Seurat_Clustering_Resolutions.tiff", plot = Clustering_UMAPs, path = "/projects/p31275/Lantz/R/Lantz_Neonate_Adult_MI/FIGURES/Seurat Clustering/", device = "tiff", dpi = 300, width = 14, height = 10, units = "in", compression = "lzw")
```

### FeaturePlot of Mac Markers
```{r}
DefaultAssay(Integrated_All_Cells) <- "SCT"

features.markers <- c("C1qa", "H2-Ab1", "Xcr1", "Siglech", "S100a9", "Plac8", "Ly6c2", "Cd3e", "Foxp3", "Gata3", "Ncr1", "Cd19", "Pecam1", "Postn", "Col3a1", "Tnnt2", "Tagln")

FeaturePlot_Cell_Markers <- FeaturePlot(Integrated_All_Cells,  features = features.markers, order = T, cols = c("gray89", "firebrick4"), min.cutoff = ("q20"), max.cutoff = ("q90"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] + 
  guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold.italic", family = "Helvetica", size = rel(2.25)),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) +
  NoLegend()
  
  FeaturePlot_Cell_Markers[[i]] <-  ggdraw() + 
  draw_plot(FeaturePlot_Cell_Markers[[i]]) +
  geom_segment(aes(x=0, y=0, xend=0.25, yend=0), arrow = arrow(length=unit(0.2, 'cm'), type = "closed")) +
  geom_segment(aes(x=0, y=0, xend=0, yend=0.3), arrow = arrow(length=unit(0.2, 'cm'), type = "closed"))
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 6)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "FeaturePlot_Cell_Markers.tiff", plot = FeaturePlot_Cell_Markers, path = "/projects/p31275/Lantz/R/Lantz_Neonate_Adult_MI/FIGURES/Seurat Clustering/", device = "tiff", dpi = 300, width = 30, height = 8, units = "in", compression = "lzw")
```

### DimPlot at Resolution 0.4
```{r}
cols.use <- c("orchid1", "firebrick3", "forestgreen", "dodgerblue4", "goldenrod", "steelblue3", "darkolivegreen3", "firebrick1", "mediumpurple3", "magenta3", "deepskyblue1", "tan1", "dimgray", "mediumpurple1", "wheat3", "deeppink1", "tan4", "firebrick4", "deeppink3", "lightcoral", "plum3", "darkolivegreen1", "springgreen3", "steelblue4", "deeppink4", "gray20")

Idents(Integrated_All_Cells) <- "Res0.4"
Dimplot_All_Cells <- DimPlot(Integrated_All_Cells, reduction = "umap", label = T, cols = cols.use)
Dimplot_All_Cells
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Dimplot_All_Cells.tiff", plot = Dimplot_All_Cells, path = "/projects/p31275/Lantz/R/Lantz_Neonate_Adult_MI/FIGURES/Seurat Clustering/", device = "tiff", dpi = 300, width = 12, height = 8, units = "in", compression = "lzw")
```

### Find Markers to Identify Each Cluster
```{r}
for(i in c(0:25)) {
  DefaultAssay(Integrated_All_Cells) <- "RNA"
  Idents(Integrated_All_Cells) <- "Res0.4"
  print(i)
  marker_i <- FindMarkers(Integrated_All_Cells, ident.1 = i, test.use = "MAST", only.pos = T, verbose = FALSE)
  marker_i <- subset(marker_i, marker_i$p_val_adj < 0.05 & abs(marker_i$avg_log2FC) > 0.5)
  marker_i$gene <- row.names(marker_i)
  write.csv(marker_i, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/Integrated_All_Cells_Markers/Integrated_All_Cells_",i,"Top_Markers.csv"))
}
```

### Rename Identities for Seurat Clusters
```{r}
Idents(Integrated_All_Cells) <- "Res0.4"

Integrated_All_Cells <- RenameIdents(Integrated_All_Cells, "0" = "Neutrophils", "1" = "Macrophages", "2" = "B Cells", "3" = "Fibroblasts", "4" = "Endothelial Cells", "5" = "Monocytes", "6" = "Fibroblasts", "7" = "T Cells", "8" = "Dendritic Cells", "9" = "Dying Cells", "10" = "NK Cells", "11" = "Endothelial Cells", "12" = "Endothelial Cells", "13" = "Cardiomyoyctes", "14" = "VSMCs", "15" = "Endothelial Cells", "16" = "Fibroblasts", "17" = "Fibroblasts", "18" = "NKT Cells", "19" = "ILCs", "20" = "Dendritic Cells", "21" = "Endothelial Cells", "22" = "Neutrophils", "23" = "Neurons", "24" = "Basophils", "25" = "Endothelial Cells")

Integrated_All_Cells$Seurat_IDs <- Idents(Integrated_All_Cells)
```

### Re-plot Dimplots
```{r}
levels <- c("Macrophages", "Monocytes", "Dendritic Cells", "Neutrophils", "B Cells", "T Cells", "NK Cells", "NKT Cells", "ILCs", "Fibroblasts", "Endothelial Cells", "Cardiomyoyctes", "VSMCs", "Neurons", "Basophils", "Dying Cells")
Integrated_All_Cells$Seurat_IDs <- factor(x = Integrated_All_Cells$Seurat_IDs, levels = levels)

cols.use <- c("dodgerblue4", "firebrick1", "deepskyblue1", "goldenrod", "deeppink4", "orchid1", "mediumpurple3", "mediumpurple4", "deeppink1", "darkolivegreen", "sienna1", "firebrick3", "darkolivegreen1", "peachpuff", "gray15", "gray")
```

### Subset out Wild Type Neonate and Adult Conditions
```{r}
#Isolate non-myeloid cells for CellChat analyses in only WILD TYPE Conditions
Idents(Integrated_All_Cells) <- "sample"
WT_All_Cells <- subset(Integrated_All_Cells, idents = c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI"))
```

### Figure S1B: Dimplot of All Cells
```{r}
Idents(WT_All_Cells) <- "Seurat_IDs"
Dimplot_All_Cells_labels <- DimPlot(WT_All_Cells, reduction = "umap", label = F, cols = cols.use) + guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-13, y=-15, xend=-7, yend=-15), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-13, y=-15, xend=-13, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.12, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)))
Dimplot_All_Cells_labels 

Dimplot_All_Cells_labels_split <- DimPlot(WT_All_Cells, reduction = "umap", split.by = "sample", label = F, cols = cols.use) + guides(color = guide_legend(override.aes = list(size=4), ncol=1))
Dimplot_All_Cells_labels_split
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Dimplot_All_Cells_labels.tiff", plot = Dimplot_All_Cells_labels, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 500, width = 9.12, height = 5.32, compression = "lzw")

ggsave(filename = "Dimplot_All_Cells_labels_split.tiff", plot = Dimplot_All_Cells_labels_split, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 300, width = 36, height = 8, units = "in", compression = "lzw")
```

### Figure S1C: Dimplot with SingleR Labels
```{r}
SingleR_IDs_Dimplot <- DimPlot(WT_All_Cells, group.by = "SingleR.labels", pt.size = 0.1, cols = c("deeppink4", "darksalmon", "black", "deepskyblue2", "sienna1", "purple4", "peachpuff", "darkolivegreen", "tan3", "dodgerblue3", "tan", "navy", "firebrick3", "darkgoldenrod1", "mediumpurple4", "deeppink1", "gray40", "gray70", "orchid3", "orchid4")) + guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-13, y=-15, xend=-7, yend=-15), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-13, y=-15, xend=-13, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(title = element_blank(), 
        axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.12, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)))

SingleR_IDs_Dimplot

SingleR_IDs_samples_Dimplot <- DimPlot(WT_All_Cells, group.by = "SingleR.labels", pt.size = 0.1, split.by = "sample", cols = c("deeppink4", "darksalmon", "black", "deepskyblue2", "sienna1", "purple4", "peachpuff", "darkolivegreen", "tan3", "dodgerblue3", "tan", "navy", "firebrick3", "darkgoldenrod1","mediumpurple4", "deeppink1", "gray40", "gray70", "orchid3", "orchid4"))
SingleR_IDs_samples_Dimplot
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "SingleR_IDs_Dimplot.tiff", plot = SingleR_IDs_Dimplot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 500, width = 9.12, height = 5.32, units = "in", compression = "lzw")

ggsave(filename = "SingleR_IDs_samples_Dimplot.tiff", plot = SingleR_IDs_samples_Dimplot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 300, width = 25, height = 8, units = "in", compression = "lzw")
```


### Figure S1D: Violin Plot of Markers across All Cell Types
```{r}
Idents(WT_All_Cells) <- "Seurat_IDs"
DefaultAssay(WT_All_Cells) <- "RNA"

features.vln <- c("C1qa", "H2-Ab1", "Xcr1", "Siglech", "S100a9", "Plac8", "Ly6c2", "Cd3e", "Foxp3", "Gata3", "Ncr1", "Cd19", "Pecam1", "Postn", "Col3a1", "Tnnt2", "Tagln")


VlnPlot <- VlnPlot(WT_All_Cells, features = features.vln, cols = cols.use, pt.size = 0.0, combine = F)
for(i in 1:(length(VlnPlot))) {
  if(i != length(VlnPlot)) {
    VlnPlot[[i]] <- VlnPlot[[i]] + ylab(features.vln[[i]]) + theme(plot.title = element_blank(), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold.italic", family = "Helvetica", size = rel(2.5)), axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks.y = element_blank()) + NoLegend()
  } else {
    VlnPlot[[i]] <- VlnPlot[[i]] + ylab(features.vln[[i]]) + theme(plot.title = element_blank(), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold.italic", family = "Helvetica", size = rel(2.5)), axis.text.x = element_text(face = "bold", family = "Helvetica", size = rel(2)), axis.text.y = element_blank(), axis.ticks.y = element_blank()) + NoLegend()
  }
}

VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol =1, rel_heights = c(rep(1, (length(VlnPlot)-1)), 2.2))
VlnPlot
```

```{r}
ggsave(filename = "VlnPlot_Cell_Markers.tiff", plot = VlnPlot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 300, width = 25, height = 35, units = "in", compression = "lzw")
```

### Subset Myeloid and Non-Myeloid Populations in WT and MerKO Experiments
```{r}
#Isolate Myeloid Cells from All Conditions for further Sub-clustering
Idents(Integrated_All_Cells) <- "Seurat_IDs"

Myeloid_Cells_All_Conditions <- subset(Integrated_All_Cells, idents = c("Macrophages", "Monocytes", "Dendritic Cells"))

Idents(WT_All_Cells) <- "Seurat_IDs"
WT_Non_Myeloid_Cells <- subset(WT_All_Cells, idents = c("Macrophages", "Monocytes", "Dendritic Cells"), invert = TRUE)
```

#### Save RDS
```{r eval=FALSE, include=FALSE}
saveRDS(Integrated_All_Cells, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Integrated_All_Cells.rds")

saveRDS(Myeloid_Cells_All_Conditions, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Myeloid_Cells_All_Conditions.rds")

saveRDS(WT_All_Cells, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/WT_All_Cells.rds")

saveRDS(WT_Non_Myeloid_Cells, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/WT_Non_Myeloid_Cells.rds")
```

# Myeloid Cells During Cardiac Regeneration and Repair
### Load Myeloid Cells Seurat Object
```{r}
Myeloid_Cells_All_Conditions <- readRDS(file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Myeloid_Cells_All_Conditions.rds")
```

### scTransform Normalization of Myeloid Cell Populations
```{r}
#Split merged filtered Seurat into individual Seurat Objects
Myeloid_Cells_All_Conditions <- SplitObject(Myeloid_Cells_All_Conditions, split.by = "sample")

#scTransform data without regressing out mitochondrial genes
for (i in 1:length(Myeloid_Cells_All_Conditions)) {
  Myeloid_Cells_All_Conditions[[i]] <- SCTransform(Myeloid_Cells_All_Conditions[[i]], vst.flavor = "v2", verbose = F)
}
```

### Integration of Normalized Datasets
#### Perform Integration for Seurat Objects normalized and scaled by sctransform of Myeloid Cell Populations
```{r}
#Prepare and Integrate SCTransformed data into an integrated Seurat Object
integ_features <- SelectIntegrationFeatures(object.list = Myeloid_Cells_All_Conditions, nfeatures = 3000)
Myeloid_Cells_All_Conditions <- PrepSCTIntegration(object.list = Myeloid_Cells_All_Conditions, anchor.features = integ_features, verbose = F)
integ_anchors <- FindIntegrationAnchors(object.list = Myeloid_Cells_All_Conditions, normalization.method = "SCT", anchor.features = integ_features, verbose = F)
Myeloid_Cells_All_Conditions <- IntegrateData(anchorset = integ_anchors, normalization.method = "SCT", verbose = F)
```

#### Perform an Integrated Analysis on Myeloid Cell Populations to confirm no batch effect
```{r}
DefaultAssay(Myeloid_Cells_All_Conditions) <- "integrated"
Myeloid_Cells_All_Conditions <- RunPCA(Myeloid_Cells_All_Conditions, npcs = 100, verbose = FALSE)
ElbowPlot(object = Myeloid_Cells_All_Conditions, ndims = 100)
```

### Seurat Clustering of Myeloid Cells
Clusters were identified using graph-based clustering approach implemented by the FindCluster function in Seurat at various resolutions to determine transcriptionally distinct populations. 
```{r}
DefaultAssay(Myeloid_Cells_All_Conditions) <- "integrated"
Myeloid_Cells_All_Conditions <- RunUMAP(Myeloid_Cells_All_Conditions, reduction = "pca", dims = 1:40, n.neighbors = 10, verbose = FALSE)
Myeloid_Cells_All_Conditions <- FindNeighbors(object = Myeloid_Cells_All_Conditions, dims = 1:40, verbose = F)
Myeloid_Cells_All_Conditions <- FindClusters(object = Myeloid_Cells_All_Conditions, resolution = c(0.4, 0.5, 0.6, 0.7, 0.9), verbose = F)

#Resolution 0.4
Idents(object = Myeloid_Cells_All_Conditions) <- "integrated_snn_res.0.4"
Myeloid_Cells_All_Conditions$Res0.4 <- Idents(object = Myeloid_Cells_All_Conditions)
p7 <- DimPlot(Myeloid_Cells_All_Conditions, reduction = "umap", label = T, label.size = 4) +NoLegend()


#Resolution 0.6
Idents(object = Myeloid_Cells_All_Conditions) <- "integrated_snn_res.0.5"
Myeloid_Cells_All_Conditions$Res0.5 <- Idents(object = Myeloid_Cells_All_Conditions)
p8 <- DimPlot(Myeloid_Cells_All_Conditions, reduction = "umap", label = T, label.size = 4)  +NoLegend()


#Resolution 0.8
Idents(object = Myeloid_Cells_All_Conditions) <- "integrated_snn_res.0.6"
Myeloid_Cells_All_Conditions$Res0.6 <- Idents(object = Myeloid_Cells_All_Conditions)
p9 <- DimPlot(Myeloid_Cells_All_Conditions, reduction = "umap", label = T, label.size = 4) + NoLegend()

#Resolution 1.0
Idents(object = Myeloid_Cells_All_Conditions) <- "integrated_snn_res.0.9"
Myeloid_Cells_All_Conditions$Res0.9 <- Idents(object = Myeloid_Cells_All_Conditions)
p10 <- DimPlot(Myeloid_Cells_All_Conditions, reduction = "umap", label = T, label.size = 4) + NoLegend()

Clustering_UMAPs <- grid.arrange(p7,p8,p9,p10, ncol = 2)
```

### Unbiased Markers
```{r message=FALSE, warning=FALSE}
for(i in c(0:16)) {
  DefaultAssay(Myeloid_Cells_All_Conditions) <- "RNA"
  Idents(Myeloid_Cells_All_Conditions) <- "Res0.6"
  print(i)
  marker_i <- FindMarkers(Myeloid_Cells_All_Conditions, ident.1 = i, test.use = "MAST", only.pos = T, verbose = FALSE)
  marker_i <- subset(marker_i, marker_i$p_val_adj < 0.05 & abs(marker_i$avg_log2FC) > 0.3)
  marker_i$gene <- row.names(marker_i)
  write.csv(marker_i, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/All_Myeloid_Cells/Top_Markers/Myeloid_Cells_All_Conditions_",i,"Top_Markers.csv"))
}
```

### Rename Identities for Seurat Clusters
```{r}
Idents(Myeloid_Cells_All_Conditions) <- "Res0.6"

Myeloid_Cells_All_Conditions <- RenameIdents(Myeloid_Cells_All_Conditions, "0" = "Ly6chi Monocytes", "1" = "C1q+ TLF+ Macrophages", "2" = "cDC2s", "3" = "Ccr2+ Macrophages", "4" = "Ly6clo Monocytes", "5" = "Arg1+ Macrophages", "6" = "Prolif. C1q+ Macrophages", "7" = "Dying Cells", "8" = "Arg1+ Macrophages", "9" = "Spp1+ Macrophages", "10" = "cDC1s", "11" = "T Cells", "12" = "Arg1+ Macrophages", "13" = "Isg Cells", "14" = "pDCs", "15" = "B Cells", "16" = "Ccr7+ DCs")
Myeloid_Cells_All_Conditions$Seurat_IDs <- Idents(Myeloid_Cells_All_Conditions)

Idents(Myeloid_Cells_All_Conditions) <- "Seurat_IDs"

Myeloid_Cells_All_Conditions <- subset(Myeloid_Cells_All_Conditions, idents = c("B Cells", "T Cells"), invert = TRUE)
```

### Subset Myeloid in WT and MerKO Experiments
```{r}
#Isolate Myeloid Cells from WT Conditions to evaluate Neonate vs Adult
Idents(Myeloid_Cells_All_Conditions) <- "sample"

Myeloid_Cells_WT <- subset(Myeloid_Cells_All_Conditions, idents = c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI"))

#Isolate Myeloid Cells from MerKO conditions to evaluate Wt vs MerKO in neonatal regeneration
Idents(Myeloid_Cells_All_Conditions) <- "sample"

Myeloid_Cells_MerKO <- subset(Myeloid_Cells_All_Conditions, idents = c("WT_P1_Sh", "WT_P1_MI", "MER_Sh", "MER_MI"))
```

#### Save RDS
```{r eval=FALSE, include=FALSE}
saveRDS(Myeloid_Cells_WT, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Myeloid_Cells_WT.rds")

saveRDS(Myeloid_Cells_MerKO, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Myeloid_Cells_MerKO.rds")

Myeloid_Cells_WT <- readRDS(file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Myeloid_Cells_WT.rds")
```


### Figure 1B: Dimplots of Myeloid Cells
```{r}
levels <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Isg Cells", "Ly6chi Monocytes", "Ly6clo Monocytes", "cDC2s", "cDC1s", "pDCs", "Ccr7+ DCs", "Dying Cells")
Myeloid_Cells_WT$Seurat_IDs <- factor(x = Myeloid_Cells_WT$Seurat_IDs, levels = levels)

levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI")
Myeloid_Cells_WT$sample <- factor(x = Myeloid_Cells_WT$sample, levels = levels)

cols.use <- c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "lightpink", "firebrick4", "tan4", "darkorange", "darkorange3", "lightgoldenrod2", "salmon1", "gray")


Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
Dimplot_Myeloid_labels <- DimPlot(Myeloid_Cells_WT, reduction = "umap", label = F, cols = cols.use) + guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), 
                                expression(bold("cDC2s")), 
                                expression(bold("cDC1s")), 
                                expression(bold("pDCs")), 
                                expression(bold("Ccr7+ DCs")), 
                                expression(bold("Dying Cells"))), 
                     values = cols.use)

Dimplot_Myeloid_labels
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Dimplot_Myeloid_labels.tiff", plot = Dimplot_Myeloid_labels, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 10, height = 6, units = "in", compression = "lzw")
```

### Myeloid Populations Across Conditions
```{r}
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
Dimplot_Myeloid_labels_split <- DimPlot(Myeloid_Cells_WT, reduction = "umap", label = F, cols = cols.use, split.by = "sample") + guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-10, y=-8, xend=-5, yend=-8), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-10, y=-8, xend=-10, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), 
                                expression(bold("cDC2s")), 
                                expression(bold("cDC1s")), 
                                expression(bold("pDCs")), 
                                expression(bold("Ccr7+ DCs")), 
                                expression(bold("Dying Cells"))), 
                     values = cols.use)

Dimplot_Myeloid_labels_split
```

### Figure 1E: FeaturePlot of Mac Markers
```{r}
DefaultAssay(Myeloid_Cells_WT) <- "SCT"

features.markers <- c("Timd4", "Lyve1", "Folr2", "C1qa", "C3ar1", "Mrc1", "Arg1", "Spp1", "Ccr2")

FeaturePlot_Cell_Markers <- FeaturePlot(Myeloid_Cells_WT,  features = features.markers, order = T, cols = c("gray89", "firebrick4"), min.cutoff = ("q20"), max.cutoff = ("q90"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] +
  theme(plot.title = element_text(hjust = 0.5, face = "bold.italic", family = "Helvetica", size = rel(2.25)),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) + NoLegend()
  
  FeaturePlot_Cell_Markers[[i]] <-  ggdraw() + 
  draw_plot(FeaturePlot_Cell_Markers[[i]]) +
  geom_segment(aes(x=0, y=0, xend=0.25, yend=0), arrow = arrow(length=unit(0.2, 'cm'), type = "closed")) +
  geom_segment(aes(x=0, y=0, xend=0, yend=0.3), arrow = arrow(length=unit(0.2, 'cm'), type = "closed"))
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 3)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Myleoid_Cell_Markers.tiff", plot = FeaturePlot_Cell_Markers, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 12, height = 12, units = "in", compression = "lzw")
```


### Figure 1C: Percent of Mac populations across each sample
```{r}
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"

Myeloid_Cells_WT_Macs <- subset(Myeloid_Cells_WT, idents = c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages"))


Prop.per.sample <- prop.table(table(Idents(Myeloid_Cells_WT_Macs), Myeloid_Cells_WT_Macs$sample), margin = 2)
Prop.per.sample
write.csv(Prop.per.sample, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/All_Myeloid_Cells/WT_Myeloid_Cell_Proportions.csv"))
```

### Figure S1E: Heatmap of top genes in each macrophage cluster
```{r message=FALSE, warning=FALSE}
levels <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages")
Myeloid_Cells_WT_Macs$Seurat_IDs <- factor(x = Myeloid_Cells_WT_Macs$Seurat_IDs, levels = levels)

seurat_IDs_myeloid <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages")

for(cell in seurat_IDs_myeloid) {
  DefaultAssay(Myeloid_Cells_WT_Macs) <- "RNA"
  Idents(Myeloid_Cells_WT_Macs) <- "Seurat_IDs"
  print(cell)
  marker_cell <- FindMarkers(Myeloid_Cells_WT_Macs, ident.1 = c(cell), test.use = "MAST", only.pos = T, verbose = FALSE)
  marker_cell <- marker_cell %>% top_n(n=25, wt = avg_log2FC)
  marker_cell$gene <- row.names(marker_cell)
  marker_cell$cell <- cell
  assign(paste0(cell, "_markers_top25"), marker_cell)
}

Myeloid_markers_top25 <- do.call("rbind", list(`C1q+ TLF+ Macrophages_markers_top25`, `Prolif. C1q+ Macrophages_markers_top25`, `Arg1+ Macrophages_markers_top25`,`Ccr2+ Macrophages_markers_top25`, `Spp1+ Macrophages_markers_top25`))

Myeloid_Heatmap <- DoHeatmap(Myeloid_Cells_WT_Macs, assay = "SCT", group.by = "Seurat_IDs", features = Myeloid_markers_top25$gene, group.colors = c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "firebrick4", "tan4"),  draw.lines = FALSE) +
  theme(legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.text.align = 0) +
  scale_fill_gradientn(colors = c("dodgerblue4", "white", "firebrick4")) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+")~bold(M*Phi*s)),
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes))), 
                                values = c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "firebrick4", "tan4"))
Myeloid_Heatmap
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Myeloid_Heatmap.tiff", plot = Myeloid_Heatmap, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 10, height = 14, units = "in", compression = "lzw")
```

### Feature Plot of MerTK Expression across WT Clusters
```{r}
DefaultAssay(Myeloid_Cells_WT) <- "RNA"

features.markers <- c("Mertk", "Ly6c2", "Ccr2", "Adgre1", "Lyve1", "Folr2")

FeaturePlot_Cell_Markers <- FeaturePlot(Myeloid_Cells_WT,  features = features.markers, order = T, cols = c("gray89", "firebrick4"), min.cutoff = ("q10"), max.cutoff = ("q90"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] +
  theme(plot.title = element_text(hjust = 0.5, face = "bold.italic", family = "Helvetica", size = rel(2.25)),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) + NoLegend()
  
  FeaturePlot_Cell_Markers[[i]] <-  ggdraw() + 
  draw_plot(FeaturePlot_Cell_Markers[[i]]) +
  geom_segment(aes(x=0, y=0, xend=0.25, yend=0), arrow = arrow(length=unit(0.2, 'cm'), type = "closed")) +
  geom_segment(aes(x=0, y=0, xend=0, yend=0.3), arrow = arrow(length=unit(0.2, 'cm'), type = "closed"))
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 6)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```

### Figure x: Expression of TBXAS1 in Neonatal and Adult Macrophage Populations
```{r}
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
DefaultAssay(Myeloid_Cells_WT) <- "RNA"

Tbxas1_Dotplot <- DotPlot(Myeloid_Cells_WT, assay = "RNA", features = c("Tbxas1", "Pla2g4a", "Pla2g7", "Pla2g15"), group.by = "Seurat_IDs", cols = c("lightgrey", "firebrick4")) + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, face = "bold.italic", size = 24), axis.title.x = element_blank(), axis.text.y = element_text(face = "bold", size = 24), axis.title.y = element_blank(), text = element_text(size=16, family="Arial"), legend.position = "bottom", legend.title = element_blank(), axis.line = element_line(linewidth = rel(2))) + coord_flip()
Tbxas1_Dotplot
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Tbxas1_Dotplot.tiff", plot = Tbxas1_Dotplot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 15, height = 5, units = "in", compression = "lzw")
```

```{r}
features.vln <- c("Tbxas1", "Pla2g15")

VlnPlot <- VlnPlot(Myeloid_Cells_WT, features = features.vln, split.by = "sample", group.by = "Seurat_IDs", idents = c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages"), pt.size = 0.1, cols = c("gray70", "deepskyblue2", "tomato", "firebrick3"), combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(2)), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.y = element_blank(), axis.ticks.y = element_blank()) + NoLegend()
}

Tbxas1_WT_VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol = 1)
Tbxas1_WT_VlnPlot
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Tbxas1_WT_VlnPlot.tiff", plot = Tbxas1_WT_VlnPlot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 14, height = 10, units = "in", compression = "lzw")
```

```{r}
features.vln <- c("Hif1a", "Igf1")

VlnPlot <- VlnPlot(Myeloid_Cells_WT, features = features.vln, split.by = "sample", group.by = "Seurat_IDs", idents = c("C1q+ TLF+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages"), pt.size = 0.1, cols = c("gray70", "deepskyblue2", "tomato", "firebrick3"), combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(2)), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.y = element_blank(), axis.ticks.y = element_blank()) + NoLegend()
}

HIF_WT_VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol = 2)
HIF_WT_VlnPlot
```

```{r}
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
C1q_WT_Macrophages <- subset(Myeloid_Cells_WT, idents = c("C1q+ TLF+ Macrophages"))

DefaultAssay(C1q_WT_Macrophages) <- "SCT"

levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI")
C1q_WT_Macrophages$sample <- factor(x = C1q_WT_Macrophages$sample, levels = levels)

features.ridge <- c("Chd4", "H3f3a", "Hdac1", "Crebbp")

Idents(C1q_WT_Macrophages) <- "sample"
Ridge_Plot_HIF <- RidgePlot(C1q_WT_Macrophages, features = features.ridge, same.y.lims = T, log = T, cols = c("dodgerblue", "dodgerblue4", "tomato", "firebrick4"), sort = F, combine = F)

for(i in 1:length(Ridge_Plot_HIF)) {
  Ridge_Plot_HIF[[i]] <- Ridge_Plot_HIF[[i]] + labs(title = features.ridge[[i]], x = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(2)), axis.title.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3), hjust = 0.5), axis.title.y = element_blank(), axis.text.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.y =  element_blank(), axis.ticks.y = element_blank(), axis.ticks.x = element_line(color = "black", linewidth = 1), axis.line = element_line(color = "black", linewidth = 1)) + NoLegend()
}


Ridge_Plot_HIF <- cowplot::plot_grid(plotlist = Ridge_Plot_HIF, ncol = 2)
Ridge_Plot_HIF
```


### Figure S1G: Biological Pathway Scoring
```{r}
AC_Clearance <- read.csv("/projects/p31275/Lantz/R/Human_Mac_Metabolism/AC_Clearance_mouse.csv", header=TRUE)
features.module <- list(AC_Clearance$name)
Myeloid_Cells_WT <- AddModuleScore(Myeloid_Cells_WT, features = features.module, name = "AC_Clearance")

Efferocytosis <- read.csv("/projects/p31275/Lantz/R/Human_Mac_Metabolism/Efferocytosis_mouse.csv", header=TRUE)
features.module <- list(Efferocytosis$name)
Myeloid_Cells_WT <- AddModuleScore(Myeloid_Cells_WT, features = features.module, name = "Efferocytosis")

Phagocytosis <- read.csv("/projects/p31275/Lantz/R/Human_Mac_Metabolism/Phagocytosis_mouse.csv", header=TRUE)
features.module <- list(Phagocytosis$name)
Myeloid_Cells_WT <- AddModuleScore(Myeloid_Cells_WT, features = features.module, name = "Phagocytosis")

Endocytosis <- read.csv("/projects/p31275/Lantz/R/Human_Mac_Metabolism/Endocytosis_mouse.csv", header=TRUE)
features.module <- list(Endocytosis$name)
Myeloid_Cells_WT <- AddModuleScore(Myeloid_Cells_WT, features = features.module, name = "Endocytosis")

Collagen_Metabolism <- read.csv("/projects/p31275/Lantz/R/Human_Mac_Metabolism/Collagen_Metabolism_mouse.csv", header=TRUE)
features.module <- list(Collagen_Metabolism$name)
Myeloid_Cells_WT <- AddModuleScore(Myeloid_Cells_WT, features = features.module, name = "Collagen_Metabolism")

Myeloid_Leukocyte_Migration <- read.csv("/projects/p31275/Lantz/R/Human_Mac_Metabolism/Myeloid_Leukocyte_Migration_mouse.csv", header=TRUE)
features.module <- list(Myeloid_Leukocyte_Migration$name)
Myeloid_Cells_WT <- AddModuleScore(Myeloid_Cells_WT, features = features.module, name = "Myeloid_Leukocyte_Migration")

Elastin_Metabolism <- read.csv("/projects/p31275/Lantz/R/Human_Mac_Metabolism/Elastin_Metabolism_mouse.csv", header=TRUE)
features.module <- list(Elastin_Metabolism$name)
Myeloid_Cells_WT <- AddModuleScore(Myeloid_Cells_WT, features = features.module, name = "Elastin_Metabolism")

Response_to_IL1 <- read.csv("/projects/p31275/Lantz/R/Human_Mac_Metabolism/Response_to_IL1_mouse.csv", header=TRUE)
features.module <- list(Response_to_IL1$name)
Myeloid_Cells_WT <- AddModuleScore(Myeloid_Cells_WT, features = features.module, name = "Response_to_IL1")
head(x = Myeloid_Cells_WT[])
```

```{r}
FeaturePlot_Cell_Markers <- FeaturePlot(Myeloid_Cells_WT, features = c("Efferocytosis1", "Myeloid_Leukocyte_Migration1", "Collagen_Metabolism1", "AC_Clearance1", "Response_to_IL11", "Elastin_Metabolism1"), order = T,  min.cutoff = ("q70"), max.cutoff = ("q98"), cols = c("gray90", "firebrick4"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] + 
  guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  theme(plot.title = element_blank(),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) +
  NoLegend()
  
  FeaturePlot_Cell_Markers[[i]] <-  ggdraw() + 
  draw_plot(FeaturePlot_Cell_Markers[[i]]) +
  geom_segment(aes(x=0, y=0, xend=0.25, yend=0), arrow = arrow(length=unit(0.2, 'cm'), type = "closed")) +
  geom_segment(aes(x=0, y=0, xend=0, yend=0.3), arrow = arrow(length=unit(0.2, 'cm'), type = "closed"))
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, nrow = 2)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "FeaturePlot_Pathways_Myeloid.tiff", plot = FeaturePlot_Cell_Markers, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 12, height = 8, units = "in", compression = "lzw")
```

### Figure S1H: Violin Plot of Cell Markers and Previously Discovered Regenerative Proteins
```{r}
levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI")
Myeloid_Cells_WT$sample <- factor(x = Myeloid_Cells_WT$sample, levels = levels)

features.vln <- c("C1qa", "Arg1", "Ccr2", "Spp1", "Tgfb1", "Ccl24", "Igf2bp3")

DefaultAssay(Myeloid_Cells_WT) <- "RNA"
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"

VlnPlot <- VlnPlot(Myeloid_Cells_WT, features = features.vln, split.by = "sample", group.by = "Seurat_IDs", idents = c("C1q+ TLF+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages"), pt.size = 0.01, cols = c("gray75", "gray25", "orange", "darkorange3"), combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(1.5)), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.5)), axis.text.x = element_blank(), axis.text.y = element_text(face = "bold", family = "Helvetica", size = rel(1.5))) + NoLegend()
}

Regenerative_VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, nrow = 1)
Regenerative_VlnPlot
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Regenerative_VlnPlot.tiff", plot = Regenerative_VlnPlot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 21, height = 3, units = "in", compression = "lzw")
```

### Figure 2G: Violin Plot of Cell Markers and Previously Discovered Regenerative Proteins
```{r}
levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI")
Myeloid_Cells_WT$sample <- factor(x = Myeloid_Cells_WT$sample, levels = levels)

features.vln <- c("Mertk", "Axl")

DefaultAssay(Myeloid_Cells_WT) <- "RNA"
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"

VlnPlot <- VlnPlot(Myeloid_Cells_WT, features = features.vln, split.by = "sample", group.by = "Seurat_IDs", idents = c("C1q+ TLF+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages"), pt.size = 0.01, cols = c("gray75", "gray25", "orange", "darkorange3"), combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(1.5)), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.5)), axis.text.x = element_blank(), axis.text.y = element_text(face = "bold", family = "Helvetica", size = rel(1.5))) + NoLegend()
}

Regenerative_VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, nrow = 1)
Regenerative_VlnPlot
```

# Differential Gene Expression
### Make ID class for cluster and sample
```{r}
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
Myeloid_Cells_WT$cluster_sample <- paste(Myeloid_Cells_WT$Seurat_IDs, Myeloid_Cells_WT$sample, sep = "_")
```

### Determine DEgenes for each cluster after MI in neonatal and adult mice
#### Neonate MI vs Adut MI
```{r message=FALSE, warning=FALSE}
seurat_IDs_myeloid <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Ly6chi Monocytes", "Ly6clo Monocytes")

seurat_IDs_myeloid <- c("Dying Cells")

for(cell in seurat_IDs_myeloid) {
  DefaultAssay(Myeloid_Cells_WT) <- "RNA"
  Idents(Myeloid_Cells_WT) <- "cluster_sample"
  print(cell)
  marker_cell <- FindMarkers(Myeloid_Cells_WT, ident.1 = paste0(cell,"_WT_P1_MI"), ident.2 = paste0(cell,"_WT_Ad_MI"), test.use = "MAST", verbose = FALSE)
  marker_cell <- subset(marker_cell, marker_cell$p_val_adj < 0.05 & abs(marker_cell$avg_log2FC) > 0.3)
  marker_cell$gene <- row.names(marker_cell)
  write.csv(marker_cell, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/All_Myeloid_Cells/WT_P1vAD_MI/DEGs_P1MIv8wMI_",cell,".csv"))
}
```

#### Adult Sham vs Neonate Sham
```{r message=FALSE, warning=FALSE}
for(cell in seurat_IDs_myeloid) {
  DefaultAssay(Myeloid_Cells_WT) <- "RNA"
  Idents(Myeloid_Cells_WT) <- "cluster_sample"
  print(cell)
  marker_cell <- FindMarkers(Myeloid_Cells_WT, ident.1 = paste0(cell,"_WT_Ad_Sh"), ident.2 = paste0(cell,"_WT_P1_Sh"), test.use = "MAST", verbose = FALSE)
  marker_cell <- subset(marker_cell, marker_cell$p_val_adj < 0.05 & abs(marker_cell$avg_log2FC) > 0.3)
  marker_cell$gene <- row.names(marker_cell)
  write.csv(marker_cell, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/All_Myeloid_Cells/WT_P1vAD_Sham/DEGs_P1ShvADSh_",cell,".csv"))
}
```

#### Adult Sham vs Neonate Sham
```{r message=FALSE, warning=FALSE}
for(cell in seurat_IDs_myeloid) {
  DefaultAssay(Myeloid_Cells_WT) <- "RNA"
  Idents(Myeloid_Cells_WT) <- "cluster_sample"
  print(cell)
  marker_cell <- FindMarkers(Myeloid_Cells_WT, ident.1 = paste0(cell,"_WT_Ad_MI"), ident.2 = "C1q+ TLF+ Macrophages_WT_P1_MI", test.use = "MAST", verbose = FALSE)
  marker_cell <- subset(marker_cell, marker_cell$p_val_adj < 0.05 & abs(marker_cell$avg_log2FC) > 0.3)
  marker_cell$gene <- row.names(marker_cell)
  write.csv(marker_cell, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/All_Myeloid_Cells/WT_P1_C1qvAD_MI/DEGs_P1MI_C1q_vADMI_",cell,".csv"))
}
```

#### Read in DEGenes
```{r}
C1q_MI_DEGenes <- read.csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/All_Myeloid_Cells/WT_P1vAD_MI/DEGs_P1MIv8wMI_C1q+ TLF+ Macrophages.csv")
rownames(C1q_MI_DEGenes) <- C1q_MI_DEGenes$X
```

### Figure 1G: Volcano Plot of DEGenes in C1q Macs
```{r}
keyvals <- ifelse(
     C1q_MI_DEGenes$avg_log2FC > 0.3, 'gray20',
      ifelse(C1q_MI_DEGenes$avg_log2FC < -0.3, 'darkorange3',
        'gray75'))
  keyvals[is.na(keyvals)] <- 'gray75'
  names(keyvals)[keyvals == 'gray20'] <- "P1 pMI"
  names(keyvals)[keyvals == 'gray75'] <- 'No Significance'
  names(keyvals)[keyvals == 'darkorange3'] <- "8w pMI"

lab_italics <- paste0("bolditalic('", rownames(C1q_MI_DEGenes), "')")

selectLab_italics = paste0("bolditalic('",
    c('mt-Nd3', 'mt-Nd2', 'mt-Nd4l', 'Rpl30', 'Rps12', 'Egr1', 'Cebpb', 'Jun', 'Klf2', 'Hpgd', 'Tgfb1', 'Cx3cr1', 'Il1b', 'Tnf', 'Runx1', 'Slc25a3', 'Igf2bp3', 'Mrc1', 'Nlrp3', 'Kras', 'Idh2', 'Nfkbiz', 'Ptgs1', 'Spp1', 'Ccl8', 'Ccl6', 'Eno1', 'Arg1', 'Fn1', 'Aldoa', 'Trem2', 'Cd74', 'Pkm', 'H2-D1', 'Hexa', 'Ecm1', 'S100a6', 'Gpx3', 'Alox5ap', 'Gpx1', 'Ctsb', 'Ctsd', 'Gas6'), "')")

Volcano_1 <- EnhancedVolcano(C1q_MI_DEGenes,
    title = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s))),
    titleLabSize = 40,
    subtitle = element_blank(),
    subtitleLabSize = 5,
    lab = lab_italics,
    selectLab = selectLab_italics,
    x = 'avg_log2FC',
    y = 'p_val_adj',
    xlab = bquote(~Log[2]~ 'Fold Change'),
    FCcutoff = 0.3, 
    pCutoff = NA,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 6.0,
    labSize = 10.0,
    labFace = 'bold',
    boxedLabels = T,
    parseLabels = T,
    colAlpha = 0.5,
    col = c("grey50", "darkorange3"),
    legendPosition = 'right',
    legendLabSize = 20,
    gridlines.major = F,
    gridlines.minor = F,
    colCustom = keyvals,
    drawConnectors = T,
    widthConnectors = 1.0,
    colConnectors = "black",
    max.overlaps = Inf, 
    arrowheads = F, 
    raster = T) + guides(color = guide_legend(override.aes = list(size = 12), ncol = 1, byrow = T)) + 
  scale_x_continuous(breaks=seq(-2,2,by=1)) +
  scale_color_manual(labels = c(expression(bold("8w pMI")), 
                                expression(bold("P1 pMI")), 
                                expression(bold("No Significance"))), values =  c("darkorange3", "gray25", "gray75")) +
  theme(legend.text.align = 0, 
        legend.spacing.y = unit(0.8, "cm"), 
        plot.title = element_text(hjust = 0.5), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(1.7)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(1.7)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.3)))
Volcano_1
```

```{r eval=FALSE, include=FALSE}
ggsave(filename = "C1q_MI_DEGenes_Volcano.tiff", plot = Volcano_1, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 20, height = 10, units = "in", compression = "lzw")
```

#### Read in DEGenes
```{r}
C1q_SH_DEGenes <- read.csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/All_Myeloid_Cells/WT_P1vAD_Sham/DEGs_P1ShvADSh_C1q+ TLF+ Macrophages.csv")
rownames(C1q_SH_DEGenes) <- C1q_SH_DEGenes$X
```

### Figure S1G: Volcano Plot of DEGenes in C1q Macrophages after Sham
```{r}
keyvals <- ifelse(
     C1q_SH_DEGenes$avg_log2FC > 0.3,'darkorange3' ,
      ifelse(C1q_SH_DEGenes$avg_log2FC < -0.3, 'gray20',
        'gray75'))
  keyvals[is.na(keyvals)] <- 'gray75'
  names(keyvals)[keyvals == 'gray20'] <- "P1 pMI"
  names(keyvals)[keyvals == 'gray75'] <- 'No Significance'
  names(keyvals)[keyvals == 'darkorange3'] <- "8w pMI"

lab_italics <- paste0("bolditalic('", rownames(C1q_SH_DEGenes), "')")

selectLab_italics = paste0("bolditalic('", 
   c('Fn1', 'H2-Aa', 'H2-Ab1', 'S100a6', 'S100a4', 'H2-Eb1', 'Cd74', 'Arg1', 'Ifitm3', 'Ecm1', 'Ccl6', 'Ccr2', 'Spp1', 'Anxa2', 'Alox5ap', 'Gpx1', 'Acly', 'Fcrls', 'Pf4', 'Igfbp4', 'Mrc1', 'Cx3cr1', 'Gas6', 'Hpgds', 'Hpgd', 'Aif1', 'Slc9a9', 'Stab1', 'Ccl12', 'Lst1', 'Idh2', 'Hmox1', 'Eif1a', 'Igf2bp3', 'Cenpa', 'Paox'), "')")

Volcano_1 <- EnhancedVolcano(C1q_SH_DEGenes,
    title = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s))),
    titleLabSize = 40,
    subtitle = element_blank(),
    subtitleLabSize = 5,
    lab = lab_italics,
    selectLab = selectLab_italics,
    x = 'avg_log2FC',
    y = 'p_val_adj',
    xlab = bquote(~Log[2]~ 'Fold Change'),
    FCcutoff = 0.3, 
    pCutoff = NA,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 6.0,
    labSize = 10.0,
    labFace = 'bold',
    boxedLabels = T,
    parseLabels = T,
    colAlpha = 0.5,
    col = c("grey50", "darkorange3"),
    legendPosition = 'right',
    legendLabSize = 20,
    gridlines.major = F,
    gridlines.minor = F,
    colCustom = keyvals,
    drawConnectors = T,
    widthConnectors = 1.0,
    colConnectors = "black",
    max.overlaps = Inf, 
    arrowheads = F, 
    raster = T) + guides(color = guide_legend(override.aes = list(size = 12), ncol = 1, byrow = T)) + 
  scale_x_continuous(breaks=seq(-2,2,by=1)) +
  scale_color_manual(labels = c(expression(bold("8w pMI")), 
                                expression(bold("P1 pMI")), 
                                expression(bold("No Significance"))), values =  c("darkorange3", "gray25", "gray75")) +
  theme(legend.text.align = 0, 
        legend.spacing.y = unit(0.8, "cm"), 
        plot.title = element_text(hjust = 0.5), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(1.7)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(1.7)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.3)))
Volcano_1
```

```{r eval=FALSE, include=FALSE}
ggsave(filename = "C1q_SHAM_DEGenes_Volcano.tiff", plot = Volcano_1, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 20, height = 10, units = "in", compression = "lzw")
```

#### Read in DEGenes
```{r}
C1q_SH_DEGenes <- read.csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/All_Myeloid_Cells/WT_P1_C1qvAD_MI/DEGs_P1MI_C1q_vADMI_Arg1+ Macrophages.csv")
rownames(C1q_SH_DEGenes) <- C1q_SH_DEGenes$X
```

### Figure S1I: Volcano Plot of DEGenes in Neonatal C1q Macrophages and Adult ARG1 Macrophages after MI
```{r}
keyvals <- ifelse(
     C1q_SH_DEGenes$avg_log2FC > 0.3,'darkorange3' ,
      ifelse(C1q_SH_DEGenes$avg_log2FC < -0.3, 'gray20',
        'gray75'))
  keyvals[is.na(keyvals)] <- 'gray75'
  names(keyvals)[keyvals == 'gray20'] <- "P1 pMI"
  names(keyvals)[keyvals == 'gray75'] <- 'No Significance'
  names(keyvals)[keyvals == 'darkorange3'] <- "8w pMI"

lab_italics <- paste0("bolditalic('", rownames(C1q_SH_DEGenes), "')")

selectLab_italics = paste0("bolditalic('", 
   c('Arg1', 'Spp1', 'Fn1', 'Slpi', 'Ecm1', 'Saa3', 'Ccl6', 'Thbs1', 'Eno1', 'Hilpda', 'Ccl9', 'Pkm', 'Aldoa', 'Ltc4s', 'Ldha', 'Ccl24', 'Pfkl', 'Alox5ap', 'Idh1', 'Gpx1', 'Hif1a', 'Osm', 'Gm42418', 'Fcrls', 'Tmem176b', 'mt-Nd3', 'Jund', 'Cx3cr1', 'Aif1', 'Mrc1', 'Hpgd', 'Hpgds', 'Tgfb1', 'Igf1', 'Gas6', 'Csf1r', 'Apoe', 'C1qbp', 'C1qa', 'Lpl', 'Ptges3', 'Axl', 'Cd47', 'Igf2bp3'), "')")

Volcano_1 <- EnhancedVolcano(C1q_SH_DEGenes,
    title = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s))),
    titleLabSize = 40,
    subtitle = element_blank(),
    subtitleLabSize = 5,
    lab = lab_italics,
    selectLab = selectLab_italics,
    x = 'avg_log2FC',
    y = 'p_val_adj',
    xlab = bquote(~Log[2]~ 'Fold Change'),
    FCcutoff = 0.3, 
    pCutoff = NA,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 6.0,
    labSize = 10.0,
    labFace = 'bold',
    boxedLabels = T,
    parseLabels = T,
    colAlpha = 0.5,
    col = c("grey50", "darkorange3"),
    legendPosition = 'right',
    legendLabSize = 20,
    gridlines.major = F,
    gridlines.minor = F,
    colCustom = keyvals,
    drawConnectors = T,
    widthConnectors = 1.0,
    colConnectors = "black",
    max.overlaps = Inf, 
    arrowheads = F, 
    raster = T) + guides(color = guide_legend(override.aes = list(size = 12), ncol = 1, byrow = T)) + 
  scale_x_continuous(breaks=seq(-2,2,by=1)) +
  scale_color_manual(labels = c(expression(bold("8w pMI")), 
                                expression(bold("P1 pMI")), 
                                expression(bold("No Significance"))), values =  c("darkorange3", "gray25", "gray75")) +
  theme(legend.text.align = 0, 
        legend.spacing.y = unit(0.8, "cm"), 
        plot.title = element_text(hjust = 0.5), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(1.7)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(1.7)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.3)))
Volcano_1
```

```{r eval=FALSE, include=FALSE}
ggsave(filename = "C1q_ARG1_MI_DEGenes_Volcano.tiff", plot = Volcano_1, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/WT_Neonate_vs_Adult/", device = "tiff", dpi = 600, width = 20, height = 10, units = "in", compression = "lzw")
```

#### Thromboxane Synthesis
```{r}
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
DefaultAssay(Myeloid_Cells_WT)<- "RNA"
VlnPlot <- VlnPlot(Myeloid_Cells_WT, features = c("Pla2g15", "Pla2g4a", "Alox5", "Alox12", "Alox15", "Ptgs1", "Ptgs2", "Tbxas1", "Ptges2", "Hpgds"),  split.by = "sample", group.by = "Seurat_IDs", cols = c("dodgerblue1", "dodgerblue4", "firebrick1", "firebrick4"), idents = c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Ly6chi Monocytes", "Ly6clo Monocytes"), pt.size = 0.2, combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + theme(axis.title.y = element_blank(), axis.title.x = element_blank(), axis.text.y = element_blank(),  axis.ticks.y = element_blank()) + NoLegend()
}
VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol =5)
VlnPlot
```




# Pseudotime Analyses
## Monocle

### Convert to Cell Data Object
```{r}
levels <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Isg Cells", "Ly6chi Monocytes", "Ly6clo Monocytes", "cDC2s", "cDC1s", "pDCs", "Ccr7+ DCs", "Dying Cells")
Myeloid_Cells_WT$Seurat_IDs <- factor(x = Myeloid_Cells_WT$Seurat_IDs, levels = levels)

Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
Myeloid_Cells_WT_Monocle <- subset(Myeloid_Cells_WT, idents = c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Isg Cells", "Ly6chi Monocytes", "Ly6clo Monocytes","Dying Cells"))

Idents(Myeloid_Cells_WT_Monocle) <- "sample"
DefaultAssay(Myeloid_Cells_WT_Monocle) <- "RNA"
Myeloid_Cells_WT_Neonate <- subset(Myeloid_Cells_WT_Monocle, idents = c("WT_P1_Sh", "WT_P1_MI"))
Myeloid_Cells_WT_Adult <- subset(Myeloid_Cells_WT_Monocle, idents = c("WT_Ad_Sh", "WT_Ad_MI"))

Myeloid_Cells_WT_Neonate_cds <- as.cell_data_set(Myeloid_Cells_WT_Neonate)
Myeloid_Cells_WT_Neonate_cds <- cluster_cells(Myeloid_Cells_WT_Neonate_cds, cluster_method = "louvain")
Myeloid_Cells_WT_Neonate_cds

Myeloid_Cells_WT_Adult_cds <- as.cell_data_set(Myeloid_Cells_WT_Adult)
Myeloid_Cells_WT_Adult_cds <- cluster_cells(Myeloid_Cells_WT_Adult_cds, cluster_method = "louvain")
Myeloid_Cells_WT_Adult_cds
``` 

### Transfer Seurat Cluster Information

#### Assign Clusters Info
```{r}
#Neonate
list_cluster <- Myeloid_Cells_WT_Neonate$Seurat_IDs
Myeloid_Cells_WT_Neonate_cds@clusters$UMAP$clusters <- list_cluster

#Adult
list_cluster <- Myeloid_Cells_WT_Adult$Seurat_IDs
Myeloid_Cells_WT_Adult_cds@clusters$UMAP$clusters <- list_cluster
```

#### Assign UMAP coordinate - cell embeddings
```{r}
#Neonate
Myeloid_Cells_WT_Neonate_cds@int_colData@listData$reducedDims$UMAP <- Myeloid_Cells_WT_Neonate@reductions$umap@cell.embeddings

#Adult
Myeloid_Cells_WT_Adult_cds@int_colData@listData$reducedDims$UMAP <- Myeloid_Cells_WT_Adult@reductions$umap@cell.embeddings
```

#### Visualize Cells in Monocle Plots
```{r}
cols.use <- c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "lightpink", "firebrick4", "tan4", "gray")

plot_cells(Myeloid_Cells_WT_Neonate_cds, color_cells_by = 'cluster', label_groups_by_cluster = F, group_label_size = 5) + theme(legend.position = "right") + scale_color_manual(values = c(cols.use))

plot_cells(Myeloid_Cells_WT_Adult_cds, color_cells_by = 'cluster', label_groups_by_cluster = F, group_label_size = 5) + theme(legend.position = "right") + scale_color_manual(values = c(cols.use))
```

### Learn Trajectory Graph
```{r}
#Neonate
Myeloid_Cells_WT_Neonate_cds <- learn_graph(Myeloid_Cells_WT_Neonate_cds)
Neonate_monocle_trajectory <- plot_cells(Myeloid_Cells_WT_Neonate_cds,
           color_cells_by = "cluster",
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = T,
           label_leaves = F,
           graph_label_size = 5,
           cell_size = 0.8,
           trajectory_graph_segment_size = rel(1.5),
           trajectory_graph_color = "black")  + guides(color = guide_legend(title = element_blank(), override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.position = "right",
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), 
                                expression(bold("Dying Cells"))), 
                     values = cols.use)

Neonate_monocle_trajectory 
```
```{r}
ggsave(filename = "Neonate_monocle_trajectory.tiff", plot = Neonate_monocle_trajectory, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Pseudotime/", device = "tiff", dpi = 300, width = 6, height = 4, units = "in", compression = "lzw")
```



```{r}
#Adult
Myeloid_Cells_WT_Adult_cds <- learn_graph(Myeloid_Cells_WT_Adult_cds)
Adult_monocle_trajectory <- plot_cells(Myeloid_Cells_WT_Adult_cds,
           color_cells_by = "cluster",
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = T,
           label_leaves = F,
           graph_label_size = 5,
           cell_size = 0.8,
           trajectory_graph_segment_size = rel(1.5),
           trajectory_graph_color = "black") + guides(color = guide_legend(title = element_blank(), override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.position = "right",
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)),
                                expression(bold("Dying Cells"))), 
                     values = cols.use)

Adult_monocle_trajectory
```
```{r}
ggsave(filename = "Adult_monocle_trajectory.tiff", plot = Adult_monocle_trajectory, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Pseudotime/", device = "tiff", dpi = 300, width = 6, height = 4, units = "in", compression = "lzw")
```


### Order Cells in Pseudotime
```{r}
Neonate_monocle_trajectory <- plot_cells(Myeloid_Cells_WT_Neonate_cds,
           color_cells_by = "cluster",
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_principal_points = T,
           label_branch_points = F,
           label_roots = F,
           label_leaves = F,
           graph_label_size = 1,
           cell_size = 0.1,
           trajectory_graph_segment_size = rel(1),
           trajectory_graph_color = "black")  + guides(color = guide_legend(title = element_blank(), override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-10, y=-8, xend=-6, yend=-8), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-10, y=-8, xend=-10, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = element_blank(), y = element_blank()) +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.position = "right",
        legend.text.align = 0) + 
scale_color_manual(labels = c(expression(bolditalic("C1qa"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bold(Prolif.)~bolditalic("C1qa"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bolditalic("Cx3cr1"^"+"~TLF^"-")~bold(M*Phi*s)),expression(bolditalic("Isg"^"+")~bold(Cells)), expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), expression(bold("Inflammatory DCs")), expression(bold("cDC2s")), expression(bold("cDC1s")), expression(bold("pDCs")), expression(bold("Neutrophils")), expression(bolditalic("Ctsdc"^"+")~bold(Neutrophils))), values = cols.use)

Neonate_monocle_trajectory 
```

```{r}
# Neonate
Myeloid_Cells_WT_Neonate_cds <- order_cells(Myeloid_Cells_WT_Neonate_cds, reduction_method = "UMAP", root_pr_nodes = c("Y_169"))

pseudotime_colors <- brewer.pal(9, "BuPu")[1:9]

Neonate_monocle_pseudotime <- plot_cells(Myeloid_Cells_WT_Neonate_cds,
           color_cells_by = "pseudotime",
           label_branch_points = F,
           label_roots = F,
           label_leaves = F,
           graph_label_size = 5,
           cell_size = 0.8,
           trajectory_graph_segment_size = rel(1.5),
           trajectory_graph_color = "black") + 
  scale_color_gradientn("Pseudotime", colors = pseudotime_colors, limits = c(0,25)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.title = element_text(face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.position = "right",
        legend.text.align = 0)
Neonate_monocle_pseudotime
```
```{r}
ggsave(filename = "Neonate_monocle_pseudotime.tiff", plot = Neonate_monocle_pseudotime, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Pseudotime/", device = "tiff", dpi = 500, width = 6, height = 4, units = "in", compression = "lzw")
```

```{r}
Adult_monocle_trajectory <- plot_cells(Myeloid_Cells_WT_Adult_cds,
           color_cells_by = "cluster",
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = F,
           label_principal_points = T,
           label_leaves = F,
           graph_label_size = 2,
           cell_size = 0.8,
           trajectory_graph_segment_size = rel(1),
           trajectory_graph_color = "black") + guides(color = guide_legend(title = element_blank(), override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-10, y=-8, xend=-6, yend=-8), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-10, y=-8, xend=-10, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = element_blank(), y = element_blank()) +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.position = "right",
        legend.text.align = 0) + 
scale_color_manual(labels = c(expression(bolditalic("C1qa"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bold(Prolif.)~bolditalic("C1qa"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bolditalic("Cx3cr1"^"+"~TLF^"-")~bold(M*Phi*s)),expression(bolditalic("Isg"^"+")~bold(Cells)), expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), expression(bold("Inflammatory DCs")), expression(bold("cDC2s")), expression(bold("cDC1s")), expression(bold("pDCs")), expression(bold("Neutrophils")), expression(bolditalic("Ctsdc"^"+")~bold(Neutrophils))), values = cols.use)

Adult_monocle_trajectory
```

```{r}
# Adult
Myeloid_Cells_WT_Adult_cds <- order_cells(Myeloid_Cells_WT_Adult_cds, reduction_method = "UMAP", root_pr_nodes = c("Y_22"))


Adult_monocle_pseudotime <- plot_cells(Myeloid_Cells_WT_Adult_cds,
           color_cells_by = "pseudotime",
           label_branch_points = F,
           label_roots = F,
           label_leaves = F,
           graph_label_size = 5,
           cell_size = 0.8,
           trajectory_graph_segment_size = rel(1.5),
           trajectory_graph_color = "black") + 
  scale_color_gradientn("Pseudotime", colors = pseudotime_colors, limits = c(0,25)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.title = element_text(face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.position = "right",
        legend.text.align = 0)

Adult_monocle_pseudotime
```
```{r}
ggsave(filename = "Adult_monocle_pseudotime.tiff", plot = Adult_monocle_pseudotime, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Pseudotime/", device = "tiff", dpi = 500, width = 6, height = 4, units = "in", compression = "lzw")
```


### Cells Ordered by Monocle3 Pseudotime
```{r include=FALSE}
# Neonate
pseudotime(Myeloid_Cells_WT_Neonate_cds)
Myeloid_Cells_WT_Neonate_cds$monocle3_pseudotime <- pseudotime(Myeloid_Cells_WT_Neonate_cds)
Neonate.pseudo <- as.data.frame(colData(Myeloid_Cells_WT_Neonate_cds))

Neonate.pseudo_Macrophages <- subset(Neonate.pseudo, Seurat_IDs == c("Arg1+ TLF+ Macs", "C1qa+ TLF+ Macs", "Spp1+ Trem2+ Macs", "Proliferating Cells"))

# Adult
pseudotime(Myeloid_Cells_WT_Adult_cds)
Myeloid_Cells_WT_Adult_cds$monocle3_pseudotime <- pseudotime(Myeloid_Cells_WT_Adult_cds)
Adult.pseudo <- as.data.frame(colData(Myeloid_Cells_WT_Adult_cds))

Adult.pseudo_Macrophages <- subset(Adult.pseudo, Seurat_IDs == c("Arg1+ TLF+ Macs", "C1qa+ TLF+ Macs", "Spp1+ Trem2+ Macs", "Ly6chi Monocytes", "Ly6clo Monocytes", "Proliferating Cells"))
```

```{r}
#Neonate
Neonate_boxplot_pseudotime <-  ggplot(Neonate.pseudo_Macrophages, aes(monocle3_pseudotime, reorder(Seurat_IDs, monocle3_pseudotime, median), Seurat_IDs, fill = Seurat_IDs)) + geom_boxplot() + scale_fill_manual(values = c(cols.use[[1]], cols.use[[2]], cols.use[[3]], cols.use[[15]])) + xlab("Pseudotime") + theme(panel.background = element_rect(fill = "white"), axis.line = element_line(color = "black", linewidth = 0.75), axis.ticks = element_line(color = "black", linewidth = 0.75), axis.text = element_text(color = "black", size = 12.0, face = "bold"), axis.title.x = element_text(color = "black", size = 12.0, face = "bold"), axis.title.y = element_blank()) + NoLegend()

Neonate_boxplot_pseudotime

#Adult
Adult_boxplot_pseudotime <- ggplot(Adult.pseudo_Macrophages, aes(monocle3_pseudotime, reorder(Seurat_IDs, monocle3_pseudotime, median), Seurat_IDs, fill = Seurat_IDs)) + geom_boxplot() + scale_fill_manual(values = c(cols.use[[1]], cols.use[[2]], cols.use[[3]], cols.use[[4]], cols.use[[5]], cols.use[[15]])) + xlab("Pseudotime") + theme(panel.background = element_rect(fill = "white"), axis.line = element_line(color = "black", linewidth = 0.75), axis.ticks = element_line(color = "black", linewidth = 0.75), axis.text = element_text(color = "black", size = 12.0, face = "bold"), axis.title.x = element_text(color = "black", size = 12.0, face = "bold"), axis.title.y = element_blank()) + NoLegend()

Adult_boxplot_pseudotime
```
```{r}
ggsave(filename = "Neonate_boxplot_pseudotime.tiff", plot = Neonate_boxplot_pseudotime, path = "/projects/p31275/Lantz/R/Lantz_Neonate_Adult_MI/FIGURES/Myeloid Cells/", device = "tiff", dpi = 300, width = 8, height = 5, units = "in", compression = "lzw")

ggsave(filename = "Adult_boxplot_pseudotime.tiff", plot = Adult_boxplot_pseudotime, path = "/projects/p31275/Lantz/R/Lantz_Neonate_Adult_MI/FIGURES/Myeloid Cells/", device = "tiff", dpi = 300, width = 8, height = 5, units = "in", compression = "lzw")
```






















# Cell Communication including Cardiomyocytes
## Process Cardiomyocyte Data from Cui et al. (GSE130699)
CellRanger output data is uploaded from Miao Cui, Zhaoning Wang, Kenian Chen, Akansha M. Shah, Wei Tan, Lauren Duan, Efrain Sanchez-Ortiz, Hui Li, Lin Xu, Ning Liu, Rhonda Bassel-Duby, Eric N. Olson,
Dynamic Transcriptional Responses to Injury of Regenerative and Non-regenerative Cardiomyocytes Revealed by Single-Nucleus RNA Sequencing, Developmental Cell, Volume 53, Issue 1, 2020, Pages 102-116.e8, ISSN 1534-5807,
https://doi.org/10.1016/j.devcel.2020.02.019.

### Create Seurat Objects and Merge for QC
```{r Load and Merge}
for (file in c("P1_SH_D3", "P1_MI_D3", "P1_SH_D1", "P1_MI_D1", "P8_SH_D3", "P8_MI_D3", "P8_SH_D1", "P8_MI_D1")){
  seurat_data <- Read10X(data.dir = paste0("/projects/p31275/Lantz/R/Lantz_Olson_Cardiomyocytes/DATA/", file))
  seurat_obj <- CreateSeuratObject(counts = seurat_data, min.features = 100, project = file)
  assign(file, seurat_obj)
}

merged_CMs_PreQC <- merge(x = P1_SH_D3, y = c(P1_MI_D3, P1_SH_D1, P1_MI_D1, P8_SH_D3, P8_MI_D3, P8_SH_D1, P8_MI_D1), 
                          add.cell.ids = c("P1_SH_D3", "P1_MI_D3", "P1_SH_D1", "P1_MI_D1", "P8_SH_D3", "P8_MI_D3", "P8_SH_D1", "P8_MI_D1"))
merged_CMs_PreQC
```

### Quality Control
```{r Quality Control}
#Log Normalization of Merged Seurat
merged_CMs_PreQC$log10GenesPerUMI <- log10(merged_CMs_PreQC$nFeature_RNA) / log10(merged_CMs_PreQC$nCount_RNA)

#Mitochondrial Feature Expression Ratio
merged_CMs_PreQC$mitoRatio <- PercentageFeatureSet(merged_CMs_PreQC, pattern = "^mt-")
merged_CMs_PreQC$mitoRatio <- merged_CMs_PreQC@meta.data$mitoRatio / 100

#Ribosomal Feature Expression Ratio
merged_CMs_PreQC$riboRatio <- PercentageFeatureSet(merged_CMs_PreQC, pattern = "^Rp[ls]")
merged_CMs_PreQC$riboRatio <- merged_CMs_PreQC@meta.data$riboRatio / 100

#Create Metadata Slot in Merged Seurat
metadata <- merged_CMs_PreQC@meta.data
metadata$cells <- rownames(metadata)
metadata <- metadata %>%
  dplyr::rename(seq_folder = orig.ident,
                nUMI = nCount_RNA,
                nGene = nFeature_RNA)

#Add Identification Metadata for Each Condition
metadata$sample <- NA
metadata$sample[which(str_detect(metadata$cells, "^P1_MI_D1"))] <- "P1_MI_D1"
metadata$sample[which(str_detect(metadata$cells, "^P1_MI_D3"))] <- "P1_MI_D3"
metadata$sample[which(str_detect(metadata$cells, "^P1_SH_D1"))] <- "P1_SH_D1"
metadata$sample[which(str_detect(metadata$cells, "^P1_SH_D3"))] <- "P1_SH_D3"
metadata$sample[which(str_detect(metadata$cells, "^P8_MI_D1"))] <- "P8_MI_D1"
metadata$sample[which(str_detect(metadata$cells, "^P8_MI_D3"))] <- "P8_MI_D3"
metadata$sample[which(str_detect(metadata$cells, "^P8_SH_D1"))] <- "P8_SH_D1"
metadata$sample[which(str_detect(metadata$cells, "^P8_SH_D3"))] <- "P8_SH_D3"

merged_CMs_PreQC@meta.data <- metadata


levels <- c("P1_SH_D3", "P1_MI_D3", "P1_SH_D1", "P1_MI_D1", "P8_SH_D3", "P8_MI_D3", "P8_SH_D1", "P8_MI_D1")
merged_CMs_PreQC$sample <- factor(x = merged_CMs_PreQC$sample, levels = levels)
metadata$sample <- factor(x = metadata$sample, levels = levels)

#nCells
p1 <- metadata %>% 
  ggplot(aes(x=sample, fill=sample)) + 
  geom_bar() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("nCells") +
  NoLegend()

#nUMIs
p2 <- metadata %>% 
  ggplot(aes(color=sample, x=nUMI, fill= sample)) + 
  geom_density(alpha = 0.2) + 
  scale_x_log10() + 
  theme_classic() +
  ylab("Cell density") +
  geom_vline(xintercept = 500) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("nUMIS")

#nGenes 
p3 <- metadata %>% 
  ggplot(aes(x=sample, y=log10(nGene), fill=sample)) + 
  geom_boxplot() + 
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("nGenes") +
  NoLegend()

#mito counts ratio
p4 <- metadata %>% 
  ggplot(aes(color=sample, x=mitoRatio, fill=sample)) + 
  geom_density(alpha = 0.1) + 
  scale_x_log10() + 
  theme_classic() +
  geom_vline(xintercept = 0.3) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("Mito Ratio") +
  NoLegend()

#ribo counts ratio
p5 <- metadata %>% 
  ggplot(aes(color=sample, x=riboRatio, fill=sample)) + 
  geom_density(alpha = 0.2) + 
  scale_x_log10() + 
  theme_classic() +
  geom_vline(xintercept = 0.5) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("Ribo Ratio") +
  NoLegend()

#novelty gene
p6 <- metadata %>%
  ggplot(aes(x=log10GenesPerUMI, color = sample, fill=sample)) +
  geom_density(alpha = 0.2) +
  theme_classic() +
  geom_vline(xintercept = 0.85) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("Unique genes per UMI") +
  NoLegend()

plot1 <- grid.arrange(p1,p2,p3,p4,p5,p6, ncol = 2)
plot1

#Correlation of genes v. UMI
plot2 <- metadata %>% 
  ggplot(aes(x=nUMI, y=nGene, color=mitoRatio)) + 
  geom_point() + 
  scale_colour_gradient(low = "deepskyblue3", high = "firebrick4") +
  stat_smooth(method=lm) +
  scale_x_log10() + 
  scale_y_log10() + 
  theme_classic() +
  geom_vline(xintercept = 750) +
  geom_hline(yintercept = 250) +
  facet_wrap(~sample)

plot2
```

### Filter Low-quality Cells
```{r Filtering CMs}
#Subset high-quality cells on Determined QC metrics from above
filtered_CMs_PostQC <- subset(x = merged_CMs_PreQC, 
                          subset= (nUMI >= 750) & 
                          (nGene >= 250) & 
                          (log10GenesPerUMI > 0.85) & 
                          (riboRatio < 0.10))

#Keep genes expressed in 10 or more cells
counts <- GetAssayData(object = filtered_CMs_PostQC, slot = "counts")
nonzero <- counts > 0
keep_genes <- Matrix::rowSums(nonzero) >= 5
filtered_counts <- counts[keep_genes, ]

#Create Seurat Object after Filtering Low Quality Cells
filtered_CMs_PostQC <- CreateSeuratObject(filtered_counts, meta.data = filtered_CMs_PostQC@meta.data)
filtered_CMs_PostQC
```

### Normalization and Integration
```{r}
#Split merged filtered Seurat into individual Seurat Objects
split_seurat <- SplitObject(filtered_CMs_PostQC, split.by = "sample")

#scTransform data without regressing out mitochondrial genes
for (i in 1:length(split_seurat)) {
  split_seurat[[i]] <- SCTransform(split_seurat[[i]], verbose = F)
}

#Prepare and Integrate SCTransformed data into an integrated Seurat Object
integ_features <- SelectIntegrationFeatures(object.list = split_seurat, nfeatures = 3000)
split_seurat <- PrepSCTIntegration(object.list = split_seurat, anchor.features = integ_features, verbose = F)
integ_anchors <- FindIntegrationAnchors(object.list = split_seurat, normalization.method = "SCT", anchor.features = integ_features, verbose = F)
Cardiomyocytes_Regeneration <- IntegrateData(anchorset = integ_anchors, normalization.method = "SCT", verbose = F)

#Run PCA and UMAP
DefaultAssay(Cardiomyocytes_Regeneration) <- "integrated"
Cardiomyocytes_Regeneration <- RunPCA(Cardiomyocytes_Regeneration, npcs = 50, verbose = FALSE)
ElbowPlot(object = Cardiomyocytes_Regeneration, ndims = 50)
Cardiomyocytes_Regeneration <- RunUMAP(Cardiomyocytes_Regeneration, reduction = "pca", dims = 1:40, verbose = FALSE)
Cardiomyocytes_Regeneration <- FindNeighbors(object = Cardiomyocytes_Regeneration, dims = 1:40, verbose = F)
Cardiomyocytes_Regeneration <- FindClusters(object = Cardiomyocytes_Regeneration, resolution = c(0.1, 0.2, 0.3, 0.4, 0.5), verbose = F)

#Resolution 0.4
Idents(object = Cardiomyocytes_Regeneration) <- "integrated_snn_res.0.1"
Cardiomyocytes_Regeneration$Res0.1 <- Idents(object = Cardiomyocytes_Regeneration)
p7 <- DimPlot(Cardiomyocytes_Regeneration, reduction = "umap", label = T, label.size = 2) +NoLegend()


#Resolution 0.6
Idents(object = Cardiomyocytes_Regeneration) <- "integrated_snn_res.0.2"
Cardiomyocytes_Regeneration$Res0.2 <- Idents(object = Cardiomyocytes_Regeneration)
p8 <- DimPlot(Cardiomyocytes_Regeneration, reduction = "umap", label = T, label.size = 2)  +NoLegend()


#Resolution 0.8
Idents(object = Cardiomyocytes_Regeneration) <- "integrated_snn_res.0.3"
Cardiomyocytes_Regeneration$Res0.3 <- Idents(object = Cardiomyocytes_Regeneration)
p9 <- DimPlot(Cardiomyocytes_Regeneration, reduction = "umap", label = T, label.size = 2) + NoLegend()

#Resolution 1.0
Idents(object = Cardiomyocytes_Regeneration) <- "integrated_snn_res.0.4"
Cardiomyocytes_Regeneration$Res0.4 <- Idents(object = Cardiomyocytes_Regeneration)
p10 <- DimPlot(Cardiomyocytes_Regeneration, reduction = "umap", label = T, label.size = 2) + NoLegend()

Clustering_UMAPs <- grid.arrange(p7,p8,p9,p10, ncol = 2)
```

### Plot Clusters and Marker Genes
```{r}
# Normalize and Scale RNA slot
DefaultAssay(Cardiomyocytes_Regeneration) <- "RNA"
Cardiomyocytes_Regeneration <- NormalizeData(Cardiomyocytes_Regeneration, verbose = F)
Cardiomyocytes_Regeneration <- FindVariableFeatures(Cardiomyocytes_Regeneration, selection.method = "vst")
Cardiomyocytes_Regeneration <- ScaleData(Cardiomyocytes_Regeneration, features = rownames(Cardiomyocytes_Regeneration))


#Overall Seurat Clusters
Idents(object = Cardiomyocytes_Regeneration) <- "Res0.1"
DimPlot(Cardiomyocytes_Regeneration, label = T)

#FeaturePlot of CM Markers
DefaultAssay(Cardiomyocytes_Regeneration) <- "RNA"
FeaturePlot_Cell_Markers <- FeaturePlot(Cardiomyocytes_Regeneration,  features = c("Myh6", "Xirp2", "Atp5b", "Casc5", "Lockd", "Top2a", "Ddc", "Gm14091", "Cd68"), order = T, ncol = 4, cols = c("gray89", "firebrick4"), combine = F)
for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] + NoLegend() + theme(axis.title.y = element_blank(), axis.title.x = element_blank(), axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks.y = element_blank())
}
FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 4)
FeaturePlot_Cell_Markers
```

### Subset out Cardiomyocytes
```{r}
Idents(object = Cardiomyocytes_Regeneration) <- "Res0.1"
WT_Cardiomyocytes <- subset(Cardiomyocytes_Regeneration, idents = c("0", "1", "5", "6", "7"))
```

### Dimplot and FeaturePlots of Cardiomyocyte Clusters
```{r}
Idents(object = WT_Cardiomyocytes) <- "Res0.1"
Dimplot_CMs <- DimPlot(WT_Cardiomyocytes)
Dimplot_CMs

#FeaturePlot of Markers from Manuscript
DefaultAssay(WT_Cardiomyocytes) <- "RNA"
FeaturePlot_Cell_Markers <- FeaturePlot(WT_Cardiomyocytes,  features = c("Myh6", "Xirp2", "Atp5b", "Sod2", "Mdh2", "Mb", "Ddc", "Gm14091"), order = T, ncol = 4, cols = c("gray89", "firebrick4"), combine = F)
for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] + NoLegend() + theme(axis.title.y = element_blank(), axis.title.x = element_blank(), axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks.y = element_blank())
}
FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 4)
FeaturePlot_Cell_Markers
```

### Rename Identities to Match Clusters in Manuscript
```{r}
Idents(object = WT_Cardiomyocytes) <- "Res0.1"
WT_Cardiomyocytes <- RenameIdents(WT_Cardiomyocytes, "0" = "CM1", "1" = "CM4", "7" = "CM2", "5" = "CM3", "6" = "CM5")
WT_Cardiomyocytes$Manuscript_IDs <- Idents(WT_Cardiomyocytes)
```

### Save Cardiomyocytes RDS
```{r}
saveRDS(WT_Cardiomyocytes, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/WT_Cardiomyocytes.rds")
```

## CellChat During Regeneration
### Read in R Data Objects for Cardiomyocytes, Myeloid Cells, and Other Cells
```{r}
WT_Cardiomyocytes <- readRDS(file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/WT_Cardiomyocytes.rds")

WT_Myeloid_Cells <- readRDS(file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Myeloid_Cells_WT.rds")

WT_Non_Myeloid_Cells <- readRDS(file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/WT_Non_Myeloid_Cells.rds")
```

### Create CellChat Identities
```{r}
# Cardiomyocytes
Idents(WT_Cardiomyocytes) <- "Manuscript_IDs"
WT_Cardiomyocytes$CellChat_IDs <- Idents(WT_Cardiomyocytes)

levels <- c("CM1", "CM2", "CM3", "CM4", "CM5")
WT_Cardiomyocytes$CellChat_IDs <- factor(x = WT_Cardiomyocytes$CellChat_IDs, levels = levels)

levels <- c("P1_SH_D1", "P1_MI_D1", "P1_SH_D3", "P1_MI_D3", "P8_SH_D1", "P8_MI_D1", "P8_SH_D3", "P8_MI_D3")
WT_Cardiomyocytes$sample <- factor(x = WT_Cardiomyocytes$sample, levels = levels)

# Myeloid Cells
Idents(WT_Myeloid_Cells) <- "Seurat_IDs"
WT_Myeloid_Cells$CellChat_IDs <- Idents(WT_Myeloid_Cells)

Idents(WT_Myeloid_Cells) <- "CellChat_IDs"
WT_Myeloid_Cells_CellChat <- subset(WT_Myeloid_Cells, idents = c("Dying Cells"), invert = TRUE)

levels <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Isg Cells", "Ly6chi Monocytes", "Ly6clo Monocytes", "cDC2s", "cDC1s", "pDCs", "Ccr7+ DCs")
WT_Myeloid_Cells_CellChat$CellChat_IDs <- factor(x = WT_Myeloid_Cells_CellChat$CellChat_IDs, levels = levels)

levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI")
WT_Myeloid_Cells_CellChat$sample <- factor(x = WT_Myeloid_Cells_CellChat$sample, levels = levels)

# Non-myeloid Cells
Idents(WT_Non_Myeloid_Cells) <- "Seurat_IDs"
WT_Non_Myeloid_Cells$CellChat_IDs <- Idents(WT_Non_Myeloid_Cells)

Idents(WT_Non_Myeloid_Cells) <- "CellChat_IDs"
WT_Non_Myeloid_Cells_CellChat <- subset(WT_Non_Myeloid_Cells, idents = c("Dying Cells", "Cardiomyoyctes"), invert = TRUE)

levels <- c("Neutrophils", "B Cells", "T Cells", "NK Cells", "NKT Cells", "ILCs", "Fibroblasts", "Endothelial Cells", "VSMCs", "Neurons", "Basophils")
WT_Non_Myeloid_Cells_CellChat$CellChat_IDs <- factor(x = WT_Non_Myeloid_Cells_CellChat$CellChat_IDs, levels = levels)

levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI")
WT_Non_Myeloid_Cells_CellChat$sample <- factor(x = WT_Non_Myeloid_Cells_CellChat$sample, levels = levels)
```

### Figure 2A: Individual UMAPs
```{r}
# Cardiomyocytes
cols.use <- c("paleturquoise3", "turquoise","seashell3","aquamarine4", "aquamarine")

Idents(WT_Cardiomyocytes) <- "CellChat_IDs"
Dimplot_Cardiomyocyte_labels <- DimPlot(WT_Cardiomyocytes, reduction = "umap", label = F, cols = cols.use) + guides(color = guide_legend(override.aes = list(size=8), ncol=1)) +
  theme_void() +
  labs(x = element_blank(), y = element_blank()) +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text.align = 0) + 
  scale_color_manual(labels = c("CM1", "CM2", "CM3", "CM4", "CM5"), values = cols.use)

Dimplot_Cardiomyocyte_labels

# Myeloid Cells
cols.use <- c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "lightpink", "firebrick4", "tan4", "darkorange", "darkorange3", "lightgoldenrod2", "salmon1", "gray")

Idents(WT_Myeloid_Cells_CellChat) <- "CellChat_IDs"
Dimplot_Myeloid_labels <- DimPlot(WT_Myeloid_Cells_CellChat, reduction = "umap", label = F, cols = cols.use) + guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  labs(x = element_blank(), y = element_blank()) +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), 
                                expression(bold("cDC2s")), 
                                expression(bold("cDC1s")), 
                                expression(bold("pDCs")), 
                                expression(bold("Ccr7+ DCs"))),
                     values = cols.use)

Dimplot_Myeloid_labels

# Non-myeloid Cells


cols.use <- c("goldenrod", "deeppink4", "orchid1", "mediumpurple3", "deeppink1", "tan4", "darkolivegreen", "lightcoral", "darkolivegreen1", "tan1", "gray15")

Idents(WT_Non_Myeloid_Cells_CellChat) <- "CellChat_IDs"
Dimplot_Non_Myeloid_Cells_labels <- DimPlot(WT_Non_Myeloid_Cells_CellChat, reduction = "umap", label = F, cols = cols.use) + guides(color = guide_legend(override.aes = list(size=4), ncol=2)) +
  labs(x = element_blank(), y = element_blank()) +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.12, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)))
Dimplot_Non_Myeloid_Cells_labels 
```
```{r}
ggsave(filename = "Dimplot_Cardiomyocyte.tiff", plot = Dimplot_Cardiomyocyte_labels, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 500, width = 10, height = 8, units = "in", compression = "lzw")

ggsave(filename = "Dimplot_Myeloid.tiff", plot = Dimplot_Myeloid_labels, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 500, width = 8, height = 6, units = "in", compression = "lzw")

ggsave(filename = "Dimplot_Non_Myeloid_Cells.tiff", plot = Dimplot_Non_Myeloid_Cells_labels, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 500, width = 10, height = 8, units = "in", compression = "lzw")
```

### Merge together Cardiomyocyte, Myeloid, and Other Cell Datasets
```{r}
CellChat_All_Conditions <- merge(x = WT_Cardiomyocytes, y = c(WT_Myeloid_Cells_CellChat, WT_Non_Myeloid_Cells_CellChat))
```

### Run a PCA analysis and Clusters to Generate UMAP for Visualization Purposes
```{r}
DefaultAssay(CellChat_All_Conditions) <- "RNA"
CellChat_All_Conditions <- NormalizeData(CellChat_All_Conditions, verbose = F)
CellChat_All_Conditions <- FindVariableFeatures(CellChat_All_Conditions, selection.method = "vst")
CellChat_All_Conditions <- ScaleData(CellChat_All_Conditions, features = rownames(CellChat_All_Conditions))
CellChat_All_Conditions <- RunPCA(CellChat_All_Conditions, npcs = 50, verbose = FALSE)
ElbowPlot(object = CellChat_All_Conditions, ndims = 50)
CellChat_All_Conditions <- RunUMAP(CellChat_All_Conditions, reduction = "pca", dims = 1:40, verbose = FALSE)
CellChat_All_Conditions <- FindNeighbors(object = CellChat_All_Conditions, dims = 1:40, verbose = F)
CellChat_All_Conditions <- FindClusters(object = CellChat_All_Conditions, resolution = c(0.1, 0.2, 0.3, 0.4, 0.5), verbose = F)
```

### Figure 2A: UMAP of Merged Datasets
```{r}
levels <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Isg Cells", "Ly6chi Monocytes", "Ly6clo Monocytes", "cDC2s", "cDC1s", "pDCs", "Ccr7+ DCs", "Neutrophils", "B Cells", "T Cells", "NK Cells", "NKT Cells", "ILCs", "Fibroblasts", "Endothelial Cells", "VSMCs", "Neurons", "Basophils", "CM1", "CM2", "CM3", "CM4", "CM5")
CellChat_All_Conditions$CellChat_IDs <- factor(x = CellChat_All_Conditions$CellChat_IDs, levels = levels)

cols.use <- c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "lightpink", "firebrick4", "tan4", "darkorange", "darkorange3", "lightgoldenrod2", "salmon1","goldenrod", "deeppink3", "orchid1", "mediumpurple3", "deeppink1", "magenta3", "darkolivegreen", "maroon4", "darkolivegreen1", "tan1", "gray15", "paleturquoise3", "turquoise","seashell3","aquamarine4", "aquamarine")

Idents(CellChat_All_Conditions) <- "CellChat_IDs"
Dimplot_Regeneration_Atlas <- DimPlot(CellChat_All_Conditions, reduction = "umap", label = F, cols = cols.use) + guides(color = guide_legend(override.aes = list(size=8), ncol=1)) +
  theme_void() +
  labs(x = element_blank(), y = element_blank()) +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text.align = 0) +
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), 
                                expression(bold("cDC2s")), 
                                expression(bold("cDC1s")), 
                                expression(bold("pDCs")), 
                                expression(bolditalic("Ccr7"^"+")~bold(DCs)),
                                expression(bold("Neutrophils")), 
                                expression(bold("B Cells")), 
                                expression(bold("T Cells")), 
                                expression(bold("NK Cells")), 
                                expression(bold("NKT Cells")),
                                expression(bold("ILCs")),
                                expression(bold("Fibroblasts")),
                                expression(bold("Endothelial Cells")),
                                expression(bold("VSMCs")), 
                                expression(bold("Neurons")), 
                                expression(bold("Basophils")),
                                expression(bold("CM1")), 
                                expression(bold("CM2")), 
                                expression(bold("CM3")), 
                                expression(bold("CM4")), 
                                expression(bold("CM5"))), 
                     values = cols.use)

Dimplot_Regeneration_Atlas
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Dimplot_Regeneration_Atlas.tiff", plot = Dimplot_Regeneration_Atlas, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 300, width = 15, height = 10, units = "in", compression = "lzw")
```


### Subset out Sample for Neonated Post-MI
Note that cardiomyocytes are 1 and 3 days pMI, and myeloid/other cells are 7 days pMI.
```{r}
cols.use <- c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "lightpink", "firebrick4", "tan4", "darkorange", "darkorange3", "lightgoldenrod2", "salmon1","goldenrod", "deeppink4", "orchid1", "mediumpurple3", "deeppink1", "tan4", "darkolivegreen", "lightcoral", "darkolivegreen1", "tan1", "gray15", "paleturquoise3", "turquoise","seashell3","aquamarine4", "aquamarine")

# Neonatal Sham
Idents(CellChat_All_Conditions) <- "sample"
P1_Sham <- subset(CellChat_All_Conditions, idents = c("P1_SH_D1", "P1_SH_D3", "WT_P1_Sh"))
Idents(P1_Sham) <- "CellChat_IDs"
DimPlot(P1_Sham, reduction = "umap", label = T, label.size = 5, cols = cols.use) + NoLegend()

# Neonatal MI
Idents(CellChat_All_Conditions) <- "sample"
P1_MI <- subset(CellChat_All_Conditions, idents = c("P1_MI_D1", "P1_MI_D3", "WT_P1_MI"))
Idents(P1_MI) <- "CellChat_IDs"
DimPlot(P1_MI, reduction = "umap", label = F, label.size = 5, cols = cols.use) 

# Non-regenerative Sham
Idents(CellChat_All_Conditions) <- "sample"
Adult_Sham <- subset(CellChat_All_Conditions, idents = c("P8_SH_D1", "P8_SH_D3", "WT_Ad_Sh"))
Idents(Adult_Sham) <- "CellChat_IDs"
DimPlot(Adult_Sham, reduction = "umap", label = F, label.size = 5, cols = cols.use) 

# Non-regenerative MI
Idents(CellChat_All_Conditions) <- "sample"
Adult_MI <- subset(CellChat_All_Conditions, idents = c("P8_MI_D1", "P8_MI_D3", "WT_Ad_MI"))
Idents(Adult_MI) <- "CellChat_IDs"
DimPlot(Adult_MI, reduction = "umap", label = F, label.size = 5, cols = cols.use) 
```

```{r}
saveRDS(P1_Sham, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/P1_Sham.rds")
saveRDS(P1_MI, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/P1_MI.rds")
saveRDS(Adult_Sham, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Adult_Sham.rds")
saveRDS(Adult_MI, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Adult_MI.rds")
```

```{r}
P1_Sham <- readRDS(file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/P1_Sham.rds")
P1_MI <- readRDS(file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/P1_MI.rds")
Adult_Sham <- readRDS(file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Adult_Sham.rds")
Adult_MI <- readRDS(file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Adult_MI.rds")
```

### Create CellChat for Each Condition
```{r}
Cellchat_conditions <- c(P1_Sham, P1_MI, Adult_Sham, Adult_MI)

for (i in 1:length(Cellchat_conditions)) {
  data.input <- GetAssayData(Cellchat_conditions[[i]], assay = "RNA", slot = "data") # normalized data matrix
  labels <- Idents(Cellchat_conditions[[i]])
  meta <- data.frame(group = labels, row.names = names(labels)) # create a dataframe of the cell labels
  Cellchat_conditions[[i]] <- createCellChat(object = data.input, meta = meta, group.by = "group")
  Cellchat_conditions[[i]] <- addMeta(Cellchat_conditions[[i]], meta = meta, meta.name = "labels")
  Cellchat_conditions[[i]] <- setIdent(Cellchat_conditions[[i]], ident.use = "labels") # set "labels" as default cell identity
}

#Load Cell Chat Database
CellChatDB <- CellChatDB.mouse
showDatabaseCategory(CellChatDB)

#Apply to Meta Data
for (i in 1:length(Cellchat_conditions)) {
  Cellchat_conditions[[i]]@DB <- CellChatDB
}

# Preprocessing the expression data for cell-cell communication analysis
for (i in 1:length(Cellchat_conditions)) {
  Cellchat_conditions[[i]]@DB <- CellChatDB
  Cellchat_conditions[[i]] <- subsetData(Cellchat_conditions[[i]])
  Cellchat_conditions[[i]] <- identifyOverExpressedGenes(Cellchat_conditions[[i]])
  Cellchat_conditions[[i]] <- identifyOverExpressedInteractions(Cellchat_conditions[[i]])
  Cellchat_conditions[[i]] <- projectData(Cellchat_conditions[[i]], PPI.mouse)
  Cellchat_conditions[[i]] <- computeCommunProb(Cellchat_conditions[[i]])
  Cellchat_conditions[[i]] <- filterCommunication(Cellchat_conditions[[i]], min.cells = 5)
  Cellchat_conditions[[i]] <- computeCommunProbPathway(Cellchat_conditions[[i]])
  Cellchat_conditions[[i]] <- aggregateNet(Cellchat_conditions[[i]], remove.isolate = F)
}

# Compute Net Neutrality
for (i in 1:length(Cellchat_conditions)) {
  Cellchat_conditions[[i]] <- netAnalysis_computeCentrality(Cellchat_conditions[[i]], slot.name = "netP")
}
```

```{r}
saveRDS(Cellchat_conditions, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Cellchat_conditions.rds")
```

#### Visualize Communications of C1q+ Macrophages across conditions
```{r}
par(mfrow = c(2,2), xpd=TRUE)

for (i in 1:length(Cellchat_conditions)) {
  netVisual_circle(Cellchat_conditions[[i]]@net$weight, targets.use = "C1q+ TLF+ Macrophages", weight.scale = T, label.edge = F, title.name = "Interaction Strength", color.use = cols.use)
}
```
### Merge CellChats Together
```{r}
Cellchat_conditions <- setNames(Cellchat_conditions, c("P1 Sham", "P1 MI", "Adult Sham", "Adult MI"))

Merged_CellChat <- mergeCellChat(Cellchat_conditions, add.names = names(Cellchat_conditions))
```

```{r}
gg1 <- compareInteractions(Merged_CellChat, show.legend = F, group = c(1,2,3,4), color.use = c("dodgerblue", "dodgerblue4", "tomato", "firebrick4"))
gg2 <- compareInteractions(Merged_CellChat, show.legend = F, group = c(1,2,3,4), measure = "weight", color.use = c("dodgerblue", "dodgerblue4", "tomato", "firebrick4"))
gg1 + gg2
```
## Differential Interactions
### Figure 2B: C1q+ Macrophage Communications after MI in Neonates and Adults
```{r}
Plot_1 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, targets.use = "C1q+ TLF+ Macrophages", comparison = c(2,4), label.edge = F, title.name = "P1 MI vs P1 Sham Number", color.use = cols.use, color.edge = c("darkorange3", "gray25"))
Plot_2 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, targets.use = "C1q+ TLF+ Macrophages", measure = "weight", vertex.label.cex = 0.01, label.edge = F, comparison = c(2,4), title.name = "P1 MI vs P1 Sham Strength", color.use = cols.use, color.edge = c("darkorange3", "gray25"))


Plot_5 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, sources.use = "C1q+ TLF+ Macrophages", comparison = c(2,4), label.edge = F, title.name = "P1 MI vs P1 Sham Number", color.use = cols.use, color.edge = c("darkorange3", "gray25"))
Plot_6 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, sources.use = "C1q+ TLF+ Macrophages", measure = "weight", vertex.label.cex = 0.01, label.edge = F, comparison = c(2,4), title.name = "P1 MI vs P1 Sham Strength", color.use = cols.use, color.edge = c("darkorange3", "gray25"))
Plot_7 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, sources.use = "C1q+ TLF+ Macrophages", comparison = c(3,4), vertex.label.cex = 0.01, label.edge = F, title.name = "Adult MI vs Adult Sham Number", color.use = cols.use)
Plot_8 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, sources.use = "C1q+ TLF+ Macrophages", measure = "weight", vertex.label.cex = 0.01, label.edge = F, comparison = c(3,4), title.name = "Adult MI vs Adult Sham Strength", color.use = cols.use)
```
```{r}
Plot_9 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, targets.use = "Arg1+ Macrophages", comparison = c(1,2), vertex.label.cex = 0.01, label.edge = F, title.name = "P1 MI vs P1 Sham Number", color.use = cols.use)
Plot_2 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, targets.use = "Arg1+ Macrophages", measure = "weight", vertex.label.cex = 0.01, label.edge = F, comparison = c(1,2), title.name = "P1 MI vs P1 Sham Strength", color.use = cols.use)
Plot_3 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, targets.use = "Arg1+ Macrophages", comparison = c(3,4), vertex.label.cex = 0.01, label.edge = F, title.name = "Adult MI vs Adult Sham Number", color.use = cols.use)
Plot_4 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, targets.use = "Arg1+ Macrophages", measure = "weight", vertex.label.cex = 0.01, label.edge = F, comparison = c(3,4), title.name = "Adult MI vs Adult Sham Strength", color.use = cols.use)

Plot_5 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, sources.use = "Ccr2+ Macrophages", comparison = c(1,2), label.edge = F, title.name = "P1 MI vs P1 Sham Number", color.use = cols.use)
Plot_6 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, sources.use = "Ccr2+ Macrophages", measure = "weight", vertex.label.cex = 0.01, label.edge = F, comparison = c(1,2), title.name = "P1 MI vs P1 Sham Strength", color.use = cols.use)
Plot_7 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, sources.use = "Ccr2+ Macrophages", comparison = c(3,4), vertex.label.cex = 0.01, label.edge = F, title.name = "Adult MI vs Adult Sham Number", color.use = cols.use)
Plot_8 <- netVisual_diffInteraction(Merged_CellChat, weight.scale = T, sources.use = "Ccr2+ Macrophages", measure = "weight", vertex.label.cex = 0.01, label.edge = F, comparison = c(3,4), title.name = "Adult MI vs Adult Sham Strength", color.use = cols.use)
```

### Figure S2A: Compare the major sources and targets in 2D space
```{r}
num.link <- sapply(Cellchat_conditions, function(x) {rowSums(x@net$count) + colSums(x@net$count)-diag(x@net$count)})
weight.MinMax <- c(min(num.link), max(num.link)) # control the dot size in the different datasets
gg <- list()
for (i in 1:length(Cellchat_conditions)) {
    gg[[i]] <- netAnalysis_signalingRole_scatter(Cellchat_conditions[[i]], title = names(Cellchat_conditions)[i], weight.MinMax = weight.MinMax, show.legend = F, color.use = cols.use, do.label = F)  +
      expand_limits(y=c(0,20), x=c(0,20)) + 
      labs(x = "Outgoing Signaling", y = "Incoming Signaling") +
      theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(1.5)),
        axis.line = element_line(color = "black", size = 2))
}

Plot_2D_space <- patchwork::wrap_plots(plots = gg, nrow = 2)
Plot_2D_space
```

```{r}
ggsave(filename = "Plot_2D_space.tiff", plot = Plot_2D_space, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat", device = "tiff", dpi = 500, width = 10, height = 8, units = "in", compression = "lzw")
```

### Figure 2C and Figure S2B: Communication Pathways in C1q Macrophages after Neonatal MI
```{r}
gg1 <- rankNet(Merged_CellChat, mode = "comparison", targets.use = "C1q+ TLF+ Macrophages", stacked = T, do.stat = T, color.use = c("gray25", "darkorange3"), comparison = c(2,4), do.flip = F, title = "Incoming") +
      guides(color = guide_legend(override.aes = list(size=8), ncol=1)) +
      labs(x = element_blank(), y = "Relative Info Flow") +
      theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(3)),
        legend.text.align = 0,
        legend.spacing.y = unit(1.0, "cm"))


gg2 <- rankNet(Merged_CellChat, mode = "comparison", sources.use = "C1q+ TLF+ Macrophages", stacked = T, do.stat = T, color.use = c("gray25", "darkorange3"), comparison = c(2,4), do.flip = F, title = "Outgoing") +
      guides(color = guide_legend(override.aes = list(size=8), ncol=1)) +
      labs(x = element_blank(), y = "Relative Info Flow") +
      theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(3)),
        legend.text.align = 0,
        legend.spacing.y = unit(1.0, "cm"))
gg1
gg2
```
```{r}
ggsave(filename = "Incoming_Signaling_Pathways.tiff", plot = gg1, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat", device = "tiff", dpi = 500, width = 20, height = 5, units = "in", compression = "lzw")

ggsave(filename = "Outgoing_Signaling_Pathways.tiff", plot = gg2, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat", device = "tiff", dpi = 500, width = 20, height = 5, units = "in", compression = "lzw")
```

```{r}
gg3 <- rankNet(Merged_CellChat, mode = "comparison", targets.use = "Arg1+ Macrophages", stacked = T, do.stat = T, color.use = c("gray25", "darkorange3"), comparison = c(3,4), do.flip = F, title = "Incoming") +
      guides(color = guide_legend(override.aes = list(size=8), ncol=1)) +
      labs(x = element_blank(), y = "Relative Info Flow") +
      theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(3)),
        legend.text.align = 0,
        legend.spacing.y = unit(1.0, "cm"))


gg4 <- rankNet(Merged_CellChat, mode = "comparison", sources.use = "Arg1+ Macrophages", stacked = T, do.stat = T, color.use = c("gray25", "darkorange3"), comparison = c(3,4), do.flip = F, title = "Outgoing") +
      guides(color = guide_legend(override.aes = list(size=8), ncol=1)) +
      labs(x = element_blank(), y = "Relative Info Flow") +
      theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(3)),
        legend.text.align = 0,
        legend.spacing.y = unit(1.0, "cm"))
gg3
gg4
```
```{r}
ggsave(filename = "Incoming_Signaling_Pathways_ARG1.tiff", plot = gg3, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat", device = "tiff", dpi = 500, width = 25, height = 5, units = "in", compression = "lzw")

ggsave(filename = "Outgoing_Signaling_Pathways_ARG1.tiff", plot = gg4, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat", device = "tiff", dpi = 500, width = 25, height = 5, units = "in", compression = "lzw")
```

### Figure 2D: Heatmap of GAS Signaling Roles after Neonatal MI
```{r}
GAS_Signaling_Roles <- netAnalysis_signalingRole_network(Cellchat_conditions[[2]],  signaling = c("GAS"), width = 15, height = 3, font.size = 10, color.use = cols.use) + theme_void() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2))

GAS_Signaling_Roles
```

### Figure S2C: Heatmap of GAS Signaling Roles
```{r}
#P1 Sham
netAnalysis_signalingRole_network(Cellchat_conditions[[1]],  signaling = c("GAS"), width = 15, height = 3, font.size = 10, color.use = cols.use) + theme_void() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2))

#8w Sham
netAnalysis_signalingRole_network(Cellchat_conditions[[3]],  signaling = c("GAS"), width = 15, height = 3, font.size = 10, color.use = cols.use) + theme_void() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2))

#8w MI
netAnalysis_signalingRole_network(Cellchat_conditions[[4]],  signaling = c("GAS"), width = 15, height = 3, font.size = 10, color.use = cols.use) + theme_void() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2))
```

### Figure S2D: Cell Communication Pathways across C1q+ Macrophages and Cardiomyocyte CM4
```{r}
rankNet(Merged_CellChat, mode = "comparison", targets.use = "CM4", sources.use = c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages"), stacked = T, do.stat = T, color.use = c("gray25", "darkorange3"), comparison = c(2,4), do.flip = F, title = "Incoming") +
      guides(color = guide_legend(override.aes = list(size=8), ncol=1)) +
      labs(x = element_blank(), y = "Relative Info Flow") +
      theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(3)),
        legend.text.align = 0,
        legend.spacing.y = unit(1.0, "cm"))
```

### Figure S2E: Heatmap of Growth Pathway Signaling across C1q+ Macrophages and Cardiomyocyte CM4
```{r}
#IGF1
netAnalysis_signalingRole_network(Cellchat_conditions[[2]],  signaling = c("IGF"), width = 15, height = 3, font.size = 10, color.use = cols.use) + theme_void() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2))

netAnalysis_signalingRole_network(Cellchat_conditions[[4]],  signaling = c("IGF"), width = 15, height = 3, font.size = 10, color.use = cols.use) + theme_void() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2))

#TWEAK
netAnalysis_signalingRole_network(Cellchat_conditions[[2]],  signaling = c("TWEAK"), width = 15, height = 3, font.size = 10, color.use = cols.use) + theme_void() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2))

#SPP1
netAnalysis_signalingRole_network(Cellchat_conditions[[2]],  signaling = c("SPP1"), width = 15, height = 3, font.size = 10, color.use = cols.use) + theme_void() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2))

netAnalysis_signalingRole_network(Cellchat_conditions[[4]],  signaling = c("SPP1"), width = 15, height = 3, font.size = 10, color.use = cols.use) + theme_void() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", family = "Helvetica", size = rel(2)), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(2)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(2)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2.5)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.line = element_line(color = "black", size = 2))

```
```{r}
saveRDS(Cellchat_conditions, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Cellchat_conditions.rds")
```

### Figure 2H: Gas-MerTK LR Circle Plots
```{r}
pathways.show <- c("Prostaglandin") 

netAnalysis_contribution(Cellchat_conditions[[2]], signaling = pathways.show)
  
pairLR.use <- extractEnrichedLR(Cellchat_conditions[[2]], signaling = pathways.show, geneLR.return = FALSE)
LR.show <- pairLR.use[4,] # show one ligand-receptor pair

# Circle plot
netVisual_individual(Cellchat_conditions[[1]], signaling = pathways.show, pairLR.use = LR.show, layout = "circle", color.use = cols.use, label.edge = F, vertex.label.cex = 0.01)
netVisual_individual(Cellchat_conditions[[2]], signaling = pathways.show, pairLR.use = LR.show, layout = "circle", color.use = cols.use, vertex.label.cex = 0.01)
netVisual_individual(Cellchat_conditions[[3]], signaling = pathways.show, pairLR.use = LR.show, layout = "circle", color.use = cols.use, vertex.label.cex = 0.01)
netVisual_individual(Cellchat_conditions[[4]], signaling = pathways.show, pairLR.use = LR.show, layout = "circle", color.use = cols.use, vertex.label.cex = 0.01)

netVisual_chord_gene(Cellchat_conditions[[2]], targets.use = "C1q+ TLF+ Macrophages", signaling = pathways.show, slot.name = 'net', lab.cex = 0.8, small.gap = 0.5, title.name = paste0("Up-regulated signaling in ", names(Cellchat_conditions[[2]])))
```


```{r}
plotGeneExpression(Merged_CellChat, signaling = "Prostaglandin", split.by = "datasets", colors.ggplot = T, type = "violin")
```






## Visualization of CellChat Communications During Regeneration
### Figure 2E: GAS Signaling in Macrophage Subsets
```{r}
levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI")
Myeloid_Cells_WT$sample <- factor(x = Myeloid_Cells_WT$sample, levels = levels)

levels <- c("C1q+ TLF+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages")
Myeloid_Cells_WT$Seurat_IDs <- factor(x = Myeloid_Cells_WT$Seurat_IDs, levels = levels)


Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
DefaultAssay(Myeloid_Cells_WT)<- "RNA"

features.vln <- c("Pros1", "Gas6", "Mertk", "Axl", "Tyro3")

VlnPlot <- VlnPlot(Myeloid_Cells_WT, features = features.vln,  split.by = "sample", group.by = "Seurat_IDs", cols =  c("dodgerblue", "dodgerblue4", "tomato", "firebrick4"), idents = c("C1q+ TLF+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages"), pt.size = 0.1, combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + 
    theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(3)), 
          axis.title.x = element_blank(), 
          axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.7)),
          axis.text.y = element_text(face = "bold", family = "Helvetica", size = rel(1.4)),
          axis.text.x = element_text(family = "Helvetica", size = rel(2)),
          axis.line = element_line(linewidth = rel(2))) + 
          NoLegend() +
    scale_x_discrete(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+"~TLF^"-")~bold(M*Phi*s))))
}

VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol = 5)
VlnPlot
```
```{r}
ggsave(filename = "VlnPlot_GAS_Pathway.tiff", plot = VlnPlot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat", device = "tiff", dpi = 500, width = 25, height = 5, units = "in", compression = "lzw")
```

### Figure 2F: FeaturePlot of GAS6 Signaling Pathway
```{r}
DefaultAssay(Myeloid_Cells_WT) <- "SCT"

features.markers <- c("Mertk", "Axl", "Tyro3", "Mrc1")

FeaturePlot_Cell_Markers <- FeaturePlot(Myeloid_Cells_WT,  features = features.markers, order = T, cols = c("gray89", "firebrick4"), max.cutoff = ("q95"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] +
  theme(plot.title = element_text(hjust = 0.5, face = "bold.italic", family = "Helvetica", size = rel(2.25)),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) + NoLegend()
  
  FeaturePlot_Cell_Markers[[i]] <-  ggdraw() + 
  draw_plot(FeaturePlot_Cell_Markers[[i]]) +
  geom_segment(aes(x=0, y=0, xend=0.25, yend=0), arrow = arrow(length=unit(0.2, 'cm'), type = "closed")) +
  geom_segment(aes(x=0, y=0, xend=0, yend=0.3), arrow = arrow(length=unit(0.2, 'cm'), type = "closed"))
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 3)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Gas_Signaling_FeaturePlots.tiff", plot = FeaturePlot_Cell_Markers, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 600, width = 10, height = 4, units = "in", compression = "lzw")
```

### Figure 2F: FeaturePlot of GAS6 Signaling Pathway
```{r}
DefaultAssay(CellChat_All_Conditions) <- "SCT"

features.markers <- c("Gas6", "Pros1", "Mertk", "Axl")

FeaturePlot_Cell_Markers <- FeaturePlot(CellChat_All_Conditions,  features = features.markers, order = T, cols = c("gray89", "firebrick4"), min.cutoff = ("q5"), max.cutoff = ("q95"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] +
  theme(plot.title = element_text(hjust = 0.5, face = "bold.italic", family = "Helvetica", size = rel(2.25)),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) + NoLegend()
  
  FeaturePlot_Cell_Markers[[i]] <-  ggdraw() + 
  draw_plot(FeaturePlot_Cell_Markers[[i]]) +
  geom_segment(aes(x=0, y=0, xend=0.25, yend=0), arrow = arrow(length=unit(0.2, 'cm'), type = "closed")) +
  geom_segment(aes(x=0, y=0, xend=0, yend=0.3), arrow = arrow(length=unit(0.2, 'cm'), type = "closed"))
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 4)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Gas_Signaling_FeaturePlots_Atlas.tiff", plot = FeaturePlot_Cell_Markers, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 600, width = 12, height = 4, units = "in", compression = "lzw")
```

```{r}
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
C1q_WT_Macrophages <- subset(Myeloid_Cells_WT, idents = c("C1q+ TLF+ Macrophages"))

DefaultAssay(C1q_WT_Macrophages) <- "RNA"

levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI")
C1q_WT_Macrophages$sample <- factor(x = C1q_WT_Macrophages$sample, levels = levels)

features.ridge <- c("Mertk", "Axl", "Tyro3")

Idents(C1q_WT_Macrophages) <- "sample"
Ridge_Plot_HIF <- RidgePlot(C1q_WT_Macrophages, features = features.ridge, same.y.lims = T, log = T, cols = c("dodgerblue", "dodgerblue4", "tomato", "firebrick4"), sort = F, combine = F)

for(i in 1:length(Ridge_Plot_HIF)) {
  Ridge_Plot_HIF[[i]] <- Ridge_Plot_HIF[[i]] + labs(title = features.ridge[[i]], x = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(2)), axis.title.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3), hjust = 0.5), axis.title.y = element_blank(), axis.text.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.y =  element_blank(), axis.ticks.y = element_blank(), axis.ticks.x = element_line(color = "black", linewidth = 1), axis.line = element_line(color = "black", linewidth = 1)) + NoLegend()
}


Ridge_Plot_C1q <- cowplot::plot_grid(plotlist = Ridge_Plot_HIF, ncol = 3)
Ridge_Plot_C1q
```
```{r}
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
Arg1_WT_Macrophages <- subset(Myeloid_Cells_WT, idents = c("Arg1+ Macrophages"))

DefaultAssay(Arg1_WT_Macrophages) <- "RNA"

levels <- c("WT_P1_Sh", "WT_P1_MI", "WT_Ad_Sh", "WT_Ad_MI")
Arg1_WT_Macrophages$sample <- factor(x = Arg1_WT_Macrophages$sample, levels = levels)

features.ridge <- c("Mertk", "Axl", "Tyro3")

Idents(Arg1_WT_Macrophages) <- "sample"
Ridge_Plot_HIF <- RidgePlot(Arg1_WT_Macrophages, features = features.ridge, same.y.lims = T, log = T, cols = c("dodgerblue", "dodgerblue4", "tomato", "firebrick4"), sort = F, combine = F)

for(i in 1:length(Ridge_Plot_HIF)) {
  Ridge_Plot_HIF[[i]] <- Ridge_Plot_HIF[[i]] + labs(title = features.ridge[[i]], x = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(2)), axis.title.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3), hjust = 0.5), axis.title.y = element_blank(), axis.text.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.y =  element_blank(), axis.ticks.y = element_blank(), axis.ticks.x = element_line(color = "black", linewidth = 1), axis.line = element_line(color = "black", linewidth = 1)) + NoLegend()
}


Ridge_Plot_Arg1 <- cowplot::plot_grid(plotlist = Ridge_Plot_HIF, ncol = 3)
Ridge_Plot_Arg1
```

### Figure 2F: GAS Expression among Cardiomyocytes
```{r}
Idents(WT_Cardiomyocytes) <- "Manuscript_IDs"
DefaultAssay(WT_Cardiomyocytes)<- "RNA"
features.vln <- c("Gas6")

Gas6_Dotplot <- DotPlot(WT_Cardiomyocytes, assay = "RNA", features = c("Gas6"), group.by = "Manuscript_IDs", cols = c("lightgrey", "firebrick4")) + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, face = "bold.italic", size = 24), axis.title.x = element_blank(), axis.text.y = element_text(face = "bold", size = 24), axis.title.y = element_blank(), text = element_text(size=16, family="Arial"), legend.position = "bottom", legend.title = element_blank(), axis.line = element_line(linewidth = rel(2))) 
Gas6_Dotplot

VlnPlot <- VlnPlot(WT_Cardiomyocytes, features = features.vln,  split.by = "sample", group.by = "Manuscript_IDs", idents = c("CM4"), pt.size = 0.1, cols = c("gray80", "dodgerblue1", "gray50", "dodgerblue3", "lightgrey", "firebrick4", "lightgrey", "firebrick4"), combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(2)), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.7)), axis.text.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.y = element_text(family = "Helvetica", size = rel(1.3))) + NoLegend()
}
VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol = 1)
VlnPlot
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Gas_Signaling_VlnPlot_CMs.tiff", plot = VlnPlot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 600, width = 4, height = 4, units = "in", compression = "lzw")
```

### Figure 6A: DEGenes of CM4 during regeneration
```{r}

```

### Figure S5E: Feature Plots of Tbxas1 and Tbxa2r across all cell types
```{r}
DefaultAssay(CellChat_All_Conditions) <- "RNA"

FeaturePlot_Cell_Markers <- FeaturePlot(CellChat_All_Conditions,  features = c("Myh6", "Col3a1", "Pecam1", "Acta2", "Adgre1", "Tbxas1", "Hpgds", "Ptgs2", "Prxl2b", "Alox5ap"), order = T, cols = c("gray89", "firebrick4"), min.cutoff = ("q50"), max.cutoff = ("q90"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] + 
  guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold.italic", family = "Helvetica", size = rel(2.25)),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) +
  NoLegend()
  
  FeaturePlot_Cell_Markers[[i]] <-  ggdraw() + 
  draw_plot(FeaturePlot_Cell_Markers[[i]]) +
  geom_segment(aes(x=0, y=0, xend=0.25, yend=0), arrow = arrow(length=unit(0.2, 'cm'), type = "closed")) +
  geom_segment(aes(x=0, y=0, xend=0, yend=0.3), arrow = arrow(length=unit(0.2, 'cm'), type = "closed"))
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 5)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```

```{r eval=FALSE, include=FALSE}
ggsave(filename = "FeaturePlot_TBXAS1_All_Cells.tiff", plot = FeaturePlot_Cell_Markers, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 300, width = 30, height = 15, units = "in", compression = "lzw")
```

### Figure S5E: Feature Plots of Tbxas1 and Tbxa2r across all cell types
```{r}
DefaultAssay(CellChat_All_Conditions) <- "RNA"

FeaturePlot_Cell_Markers <- FeaturePlot(CellChat_All_Conditions,  features = c("Tbxa2r", "Ptger2", "Ptger4", "Ltb4r1", "Ptgir", "Ptgfr"), order = T, cols = c("gray89", "firebrick4"), min.cutoff = ("q50"), max.cutoff = ("q90"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] + 
  guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold.italic", family = "Helvetica", size = rel(2.25)),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) +
  NoLegend()
  
  FeaturePlot_Cell_Markers[[i]] <-  ggdraw() + 
  draw_plot(FeaturePlot_Cell_Markers[[i]]) +
  geom_segment(aes(x=0, y=0, xend=0.25, yend=0), arrow = arrow(length=unit(0.2, 'cm'), type = "closed")) +
  geom_segment(aes(x=0, y=0, xend=0, yend=0.3), arrow = arrow(length=unit(0.2, 'cm'), type = "closed"))
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 3)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "FeaturePlot_TBXA2R_All_Cells.tiff", plot = FeaturePlot_Cell_Markers, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 500, width = 10, height = 6, units = "in", compression = "lzw")
```

### Figure S5F: Violin Plot of Tbxa2r Expression Across Cardiomyocyte Populations
```{r}
Idents(WT_Cardiomyocytes) <- "Manuscript_IDs"
DefaultAssay(WT_Cardiomyocytes)<- "RNA"

cols.use <- c("paleturquoise3", "turquoise","seashell3","aquamarine4", "aquamarine")

features.vln <- c("Tbxa2r")

VlnPlot <- VlnPlot(WT_Cardiomyocytes, features = features.vln, cols = cols.use, pt.size = 0.5, combine = F)

for(i in 1:(length(VlnPlot))) {
  
    VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(2)), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.x = element_text(face = "bold", family = "Helvetica", size = rel(1.3)), axis.text.y = element_blank(), axis.ticks.y = element_blank()) + NoLegend()
  }

VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol = 1)
VlnPlot
```

### Figure S5G: DotPlot of TBXA2R Expression
```{r}
Idents(WT_Cardiomyocytes) <- "Manuscript_IDs"

WT_Cardiomyocytes_CM4 <- subset(WT_Cardiomyocytes, idents = c("CM4"))

levels <- c("P1_SH_D1", "P1_MI_D1", "P1_SH_D3", "P1_MI_D3", "P8_SH_D1", "P8_MI_D1", "P8_SH_D3", "P8_MI_D3")
WT_Cardiomyocytes_CM4$sample <- factor(x = WT_Cardiomyocytes_CM4$sample, levels = levels)

dotplot_colors <- c("gray75", "firebrick4")

Idents(WT_Cardiomyocytes_CM4) <- "sample"

Eicosanoid_Receptor_Dotplot <- DotPlot(WT_Cardiomyocytes_CM4, assay = "SCT", features = c("Tbxa2r", "Ptgir", "Ptgfr", "Ptger1", "Ptger2", "Ptger3", "Ptger4", "Ltb4r1", "Cysltr1", "Cmklr1", "Gpr18"), cols = dotplot_colors)+RotatedAxis()+coord_flip()+theme(axis.text.x = element_blank(), axis.text.y=element_text( face = "bold.italic", size=16), axis.title.y = element_blank(), axis.title.x = element_blank())
Eicosanoid_Receptor_Dotplot
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Eicosanoid_Receptor_Dotplot_CM4.tiff", plot = Eicosanoid_Receptor_Dotplot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/CellChat/", device = "tiff", dpi = 500, width = 6, height = 4, units = "in", compression = "lzw")
```


```{r}
saveRDS(CellChat_All_Conditions, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/CellChat_All_Conditions.rds")
```


# MerTK KO during Neonatal Cardiac Regeneration
```{r}
Myeloid_Cells_MerKO <- readRDS("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/SEURAT_Objects/Myeloid_Cells_MerKO.rds")

levels <- c("WT_P1_Sh", "WT_P1_MI", "MER_Sh", "MER_MI")
Myeloid_Cells_MerKO$sample <- factor(x = Myeloid_Cells_MerKO$sample, levels = levels)

Idents(Myeloid_Cells_MerKO) <- "sample"
Myeloid_Cells_MerKO_only <- subset(Myeloid_Cells_MerKO, idents = c("MER_Sh", "MER_MI"))
```

### Figure 3B: Dimplots of Myeloid Cells in MER KO Neonates
```{r}
levels <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Isg Cells", "Ly6chi Monocytes", "Ly6clo Monocytes", "cDC2s", "cDC1s", "pDCs", "Ccr7+ DCs", "Dying Cells")
Myeloid_Cells_MerKO$Seurat_IDs <- factor(x = Myeloid_Cells_MerKO$Seurat_IDs, levels = levels)

levels <- c("MER_Sh", "MER_MI")
Myeloid_Cells_MerKO$sample <- factor(x = Myeloid_Cells_MerKO$sample, levels = levels)

cols.use <- c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "lightpink", "firebrick4", "tan4", "darkorange", "darkorange3", "lightgoldenrod2", "salmon1", "gray")


Idents(Myeloid_Cells_MerKO) <- "Seurat_IDs"
Dimplot_Myeloid_labels <- DimPlot(Myeloid_Cells_MerKO, reduction = "umap", label = F, cols = cols.use) + guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), 
                                expression(bold("cDC2s")), 
                                expression(bold("cDC1s")), 
                                expression(bold("pDCs")), 
                                expression(bold("Ccr7+ DCs")), 
                                expression(bold("Dying Cells"))), 
                     values = cols.use)

Dimplot_Myeloid_labels
```

```{r eval=FALSE, include=FALSE}
ggsave(filename = "Dimplot_Myeloid_labels_MerKO_only.tiff", plot = Dimplot_Myeloid_labels, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/MerKO/", device = "tiff", dpi = 600, width = 10, height = 6, units = "in", compression = "lzw")
```

### Figure 3B: Dimplots of Myeloid Cells in MER KO Neonates
```{r}
Idents(Myeloid_Cells_MerKO) <- "Seurat_IDs"
Dimplot_Myeloid_labels <- DimPlot(Myeloid_Cells_MerKO, reduction = "umap", label = F, split.by = "sample", cols = cols.use) + guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  theme_void() +
  theme(axis.title= element_blank(),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), 
                                expression(bold("cDC2s")), 
                                expression(bold("cDC1s")), 
                                expression(bold("pDCs")), 
                                expression(bold("Ccr7+ DCs")), 
                                expression(bold("Dying Cells"))), 
                     values = cols.use)

Dimplot_Myeloid_labels
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "Dimplot_Myeloid_labels_MerKO_split.tiff", plot = Dimplot_Myeloid_labels, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/MerKO/", device = "tiff", dpi = 600, width = 16, height = 4, units = "in", compression = "lzw")
```

### Figure 3E: Percent of Mac populations across each sample
```{r}
Idents(Myeloid_Cells_MerKO) <- "Seurat_IDs"

Myeloid_Cells_MerKO_Macs <- subset(Myeloid_Cells_MerKO, idents = c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages"))


Prop.per.sample <- prop.table(table(Idents(Myeloid_Cells_MerKO_Macs), Myeloid_Cells_MerKO_Macs$sample), margin = 2)
Prop.per.sample
write.csv(Prop.per.sample, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/All_Myeloid_Cells/Myeloid_Cells_MerKO_Proportions.csv"))
```

### Make ID class for cluster and sample
```{r}
Idents(Myeloid_Cells_MerKO) <- "Seurat_IDs"
Myeloid_Cells_MerKO$cluster_sample <- paste(Myeloid_Cells_MerKO$Seurat_IDs, Myeloid_Cells_MerKO$sample, sep = "_")
```

#### WT Sham versus MerKO Sham
```{r message=FALSE, warning=FALSE}
seurat_IDs_myeloid <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Ly6chi Monocytes", "Ly6clo Monocytes")

for(cell in seurat_IDs_myeloid) {
  DefaultAssay(Myeloid_Cells_MerKO) <- "RNA"
  Idents(Myeloid_Cells_MerKO) <- "cluster_sample"
  print(cell)
  marker_cell <- FindMarkers(Myeloid_Cells_MerKO, ident.1 = paste0(cell,"_MER_Sh"), ident.2 = paste0(cell,"_WT_P1_Sh"), test.use = "MAST", verbose = FALSE)
  marker_cell <- subset(marker_cell, marker_cell$p_val_adj < 0.05 & abs(marker_cell$avg_log2FC) > 0.3)
  marker_cell$gene <- row.names(marker_cell)
  write.csv(marker_cell, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/Sham Comparison/DEGs_WTShvMERSh_",cell,".csv"))
}
```

#### WT MI versus MerKO MI
```{r message=FALSE, warning=FALSE}
seurat_IDs_myeloid <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Ly6chi Monocytes", "Ly6clo Monocytes")

for(cell in seurat_IDs_myeloid) {
  DefaultAssay(Myeloid_Cells_MerKO) <- "RNA"
  Idents(Myeloid_Cells_MerKO) <- "cluster_sample"
  print(cell)
  marker_cell <- FindMarkers(Myeloid_Cells_MerKO, ident.1 = paste0(cell,"_MER_MI"), ident.2 = paste0(cell,"_WT_P1_MI"), test.use = "MAST", verbose = FALSE)
  marker_cell <- subset(marker_cell, marker_cell$p_val_adj < 0.05 & abs(marker_cell$avg_log2FC) > 0.3)
  marker_cell$gene <- row.names(marker_cell)
  write.csv(marker_cell, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/DEGs_WTMIvMERMI_",cell,".csv"))
}
```

```{r}
Idents(Myeloid_Cells_MerKO) <- "cluster_sample"
markers <- FindMarkers(Myeloid_Cells_MerKO, ident.1 = "Arg1+ Macrophages_MER_MI", ident.2 = "Ccr2+ Macrophages_MER_MI", test.use = "MAST", verbose = FALSE)

markers <- subset(markers, markers$p_val_adj < 0.05 & abs(markers$avg_log2FC) > 0.3)

markers$gene <- row.names(markers)

write.csv(markers, file = paste0("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/DEGs_Arg1ShvC1qSh.csv"))

```

#### Macrophage Markers
```{r}
Idents(Myeloid_Cells_MerKO) <- "Seurat_IDs"
DefaultAssay(Myeloid_Cells_MerKO)<- "RNA"
features.vln <- c("C1qa", "Arg1", "Ccr2", "Spp1")

VlnPlot <- VlnPlot(Myeloid_Cells_MerKO, features = features.vln,  split.by = "sample", group.by = "Seurat_IDs", cols = c("gray75", "gray25", "firebrick1", "firebrick4"), idents = c("C1q+ TLF+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages"), pt.size = 0.1, combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(2)), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.5)), axis.text.x = element_blank(), axis.text.y = element_text(face = "bold", family = "Helvetica", size = rel(1.5))) + NoLegend()
}

MerKO_VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol = 2)
MerKO_VlnPlot
```
```{r}
ggsave(filename = "MerKO_VlnPlot_Mac_Markers.tiff", plot = MerKO_VlnPlot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/MerKO/", device = "tiff", dpi = 500, width = 8, height = 4, units = "in", compression = "lzw")
```


#### Thromboxane Synthesis
```{r}
Idents(Myeloid_Cells_MerKO) <- "Seurat_IDs"
DefaultAssay(Myeloid_Cells_MerKO)<- "SCT"
VlnPlot <- VlnPlot(Myeloid_Cells_MerKO, features = c("Pla2g15", "Pla2g4a", "Alox5", "Alox12", "Alox15", "Ptgs1", "Ptgs2", "Tbxas1", "Ptges2", "Hpgds"),  split.by = "sample", group.by = "Seurat_IDs", cols = c("dodgerblue1", "dodgerblue4", "firebrick1", "firebrick4"), idents = c("C1q+ TLF+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages"), pt.size = 0, combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + theme(axis.title.y = element_blank(), axis.title.x = element_blank(), axis.text.y = element_blank(),  axis.ticks.y = element_blank()) + NoLegend()
}
VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol =5)
VlnPlot
```

## Monocle MerTK KO

### Convert to Cell Data Object
```{r}
levels <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Isg Cells", "Ly6chi Monocytes", "Ly6clo Monocytes", "cDC2s", "cDC1s", "pDCs", "Ccr7+ DCs", "Dying Cells")
Myeloid_Cells_MerKO$Seurat_IDs <- factor(x = Myeloid_Cells_MerKO$Seurat_IDs, levels = levels)

Idents(Myeloid_Cells_MerKO) <- "Seurat_IDs"
Myeloid_Cells_MerKO_Monocle <- subset(Myeloid_Cells_MerKO, idents = c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Isg Cells", "Ly6chi Monocytes", "Ly6clo Monocytes","Dying Cells"))

Idents(Myeloid_Cells_MerKO_Monocle) <- "sample"
DefaultAssay(Myeloid_Cells_MerKO_Monocle) <- "RNA"
Myeloid_Cells_WT_Neonate <- subset(Myeloid_Cells_MerKO_Monocle, idents = c("WT_P1_Sh", "WT_P1_MI"))
Myeloid_Cells_MerKO_Neonate <- subset(Myeloid_Cells_MerKO_Monocle, idents = c("MER_Sh", "MER_MI"))

Myeloid_Cells_WT_Neonate_cds <- as.cell_data_set(Myeloid_Cells_WT_Neonate)
Myeloid_Cells_WT_Neonate_cds <- cluster_cells(Myeloid_Cells_WT_Neonate_cds, cluster_method = "louvain")
Myeloid_Cells_WT_Neonate_cds

Myeloid_Cells_MerKO_Neonate_cds <- as.cell_data_set(Myeloid_Cells_MerKO_Neonate)
Myeloid_Cells_MerKO_Neonate_cds <- cluster_cells(Myeloid_Cells_MerKO_Neonate_cds, cluster_method = "louvain")
Myeloid_Cells_MerKO_Neonate_cds
``` 

### Transfer Seurat Cluster Information

#### Assign Clusters Info
```{r}
#WT
list_cluster <- Myeloid_Cells_WT_Neonate$Seurat_IDs
Myeloid_Cells_WT_Neonate_cds@clusters$UMAP$clusters <- list_cluster

#MerKO
list_cluster <- Myeloid_Cells_MerKO_Neonate$Seurat_IDs
Myeloid_Cells_MerKO_Neonate_cds@clusters$UMAP$clusters <- list_cluster
```

#### Assign UMAP coordinate - cell embeddings
```{r}
#WT
Myeloid_Cells_WT_Neonate_cds@int_colData@listData$reducedDims$UMAP <- Myeloid_Cells_WT_Neonate@reductions$umap@cell.embeddings

#MerKO
Myeloid_Cells_MerKO_Neonate_cds@int_colData@listData$reducedDims$UMAP <- Myeloid_Cells_MerKO_Neonate@reductions$umap@cell.embeddings
```

#### Visualize Cells in Monocle Plots
```{r}
cols.use <- c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "lightpink", "firebrick4", "tan4", "gray")

plot_cells(Myeloid_Cells_WT_Neonate_cds, color_cells_by = 'cluster', label_groups_by_cluster = F, group_label_size = 5) + theme(legend.position = "right") + scale_color_manual(values = c(cols.use))

plot_cells(Myeloid_Cells_MerKO_Neonate_cds, color_cells_by = 'cluster', label_groups_by_cluster = F, group_label_size = 5) + theme(legend.position = "right") + scale_color_manual(values = c(cols.use))
```

### Learn Trajectory Graph
```{r}
#Neonate
Myeloid_Cells_WT_Neonate_cds <- learn_graph(Myeloid_Cells_WT_Neonate_cds)
Neonate_monocle_trajectory <- plot_cells(Myeloid_Cells_WT_Neonate_cds,
           color_cells_by = "cluster",
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = T,
           label_leaves = F,
           graph_label_size = 5,
           cell_size = 0.8,
           trajectory_graph_segment_size = rel(1.5),
           trajectory_graph_color = "black")  + guides(color = guide_legend(title = element_blank(), override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.position = "right",
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), 
                                expression(bold("Dying Cells"))), 
                     values = cols.use)

Neonate_monocle_trajectory 
```
```{r}
ggsave(filename = "Neonate_WT_monocle_trajectory.tiff", plot = Neonate_monocle_trajectory, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Pseudotime/", device = "tiff", dpi = 300, width = 6, height = 4, units = "in", compression = "lzw")
```



```{r}
#MerKO
Myeloid_Cells_MerKO_Neonate_cds <- learn_graph(Myeloid_Cells_MerKO_Neonate_cds)
MerKO_monocle_trajectory <- plot_cells(Myeloid_Cells_MerKO_Neonate_cds,
           color_cells_by = "cluster",
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = T,
           label_leaves = F,
           graph_label_size = 5,
           cell_size = 0.8,
           trajectory_graph_segment_size = rel(1.5),
           trajectory_graph_color = "black") + guides(color = guide_legend(title = element_blank(), override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.position = "right",
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)),
                                expression(bold("Dying Cells"))), 
                     values = cols.use)

MerKO_monocle_trajectory
```
```{r}
ggsave(filename = "MerKO_monocle_trajectory.tiff", plot = MerKO_monocle_trajectory, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Pseudotime/", device = "tiff", dpi = 300, width = 6, height = 4, units = "in", compression = "lzw")
```


### Order Cells in Pseudotime
```{r}
Neonate_monocle_trajectory <- plot_cells(Myeloid_Cells_WT_Neonate_cds,
           color_cells_by = "cluster",
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_principal_points = T,
           label_branch_points = F,
           label_roots = F,
           label_leaves = F,
           graph_label_size = 1,
           cell_size = 0.1,
           trajectory_graph_segment_size = rel(1),
           trajectory_graph_color = "black")  + guides(color = guide_legend(title = element_blank(), override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-10, y=-8, xend=-6, yend=-8), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-10, y=-8, xend=-10, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = element_blank(), y = element_blank()) +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.position = "right",
        legend.text.align = 0) + 
scale_color_manual(labels = c(expression(bolditalic("C1qa"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bold(Prolif.)~bolditalic("C1qa"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bolditalic("Cx3cr1"^"+"~TLF^"-")~bold(M*Phi*s)),expression(bolditalic("Isg"^"+")~bold(Cells)), expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), expression(bold("Inflammatory DCs")), expression(bold("cDC2s")), expression(bold("cDC1s")), expression(bold("pDCs")), expression(bold("Neutrophils")), expression(bolditalic("Ctsdc"^"+")~bold(Neutrophils))), values = cols.use)

Neonate_monocle_trajectory 
```

```{r}
# Neonate
Myeloid_Cells_WT_Neonate_cds <- order_cells(Myeloid_Cells_WT_Neonate_cds, reduction_method = "UMAP", root_pr_nodes = c("Y_169"))

pseudotime_colors <- brewer.pal(9, "BuPu")[1:9]

Neonate_monocle_pseudotime <- plot_cells(Myeloid_Cells_WT_Neonate_cds,
           color_cells_by = "pseudotime",
           label_branch_points = F,
           label_roots = F,
           label_leaves = F,
           graph_label_size = 5,
           cell_size = 0.8,
           trajectory_graph_segment_size = rel(1.5),
           trajectory_graph_color = "black") + 
  scale_color_gradientn("Pseudotime", colors = pseudotime_colors, limits = c(0,25)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.title = element_text(face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.position = "right",
        legend.text.align = 0)
Neonate_monocle_pseudotime
```
```{r}
ggsave(filename = "Neonate_WT_monocle_pseudotime.tiff", plot = Neonate_monocle_pseudotime, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Pseudotime/", device = "tiff", dpi = 500, width = 6, height = 4, units = "in", compression = "lzw")
```

```{r}
MerKO_monocle_trajectory <- plot_cells(Myeloid_Cells_MerKO_Neonate_cds,
           color_cells_by = "cluster",
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_principal_points = T,
           label_branch_points = F,
           label_roots = F,
           label_leaves = F,
           graph_label_size = 1,
           cell_size = 0.1,
           trajectory_graph_segment_size = rel(1),
           trajectory_graph_color = "black")  + guides(color = guide_legend(title = element_blank(), override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-10, y=-8, xend=-6, yend=-8), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-10, y=-8, xend=-10, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = element_blank(), y = element_blank()) +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.position = "right",
        legend.text.align = 0) + 
scale_color_manual(labels = c(expression(bolditalic("C1qa"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bold(Prolif.)~bolditalic("C1qa"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bolditalic("Arg1"^"+"~TLF^"+")~bold(M*Phi*s)), expression(bolditalic("Cx3cr1"^"+"~TLF^"-")~bold(M*Phi*s)),expression(bolditalic("Isg"^"+")~bold(Cells)), expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), expression(bold("Inflammatory DCs")), expression(bold("cDC2s")), expression(bold("cDC1s")), expression(bold("pDCs")), expression(bold("Neutrophils")), expression(bolditalic("Ctsdc"^"+")~bold(Neutrophils))), values = cols.use)

MerKO_monocle_trajectory
```

```{r}
# MerKO
Myeloid_Cells_MerKO_Neonate_cds <- order_cells(Myeloid_Cells_MerKO_Neonate_cds, reduction_method = "UMAP", root_pr_nodes = c("Y_48"))


MerKO_monocle_pseudotime <- plot_cells(Myeloid_Cells_MerKO_Neonate_cds,
           color_cells_by = "pseudotime",
           label_branch_points = F,
           label_roots = F,
           label_leaves = F,
           graph_label_size = 5,
           cell_size = 0.8,
           trajectory_graph_segment_size = rel(1.5),
           trajectory_graph_color = "black") + 
  scale_color_gradientn("Pseudotime", colors = pseudotime_colors, limits = c(0,25)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.title = element_text(face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.position = "right",
        legend.text.align = 0)

MerKO_monocle_pseudotime
```
```{r}
ggsave(filename = "MerKO_monocle_pseudotime.tiff", plot = MerKO_monocle_pseudotime, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/Pseudotime/", device = "tiff", dpi = 500, width = 6, height = 4, units = "in", compression = "lzw")
```

#### Read in DEGenes
```{r}
C1q_MI_DEGenes <- read.csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/DEGs_WTMIvMERMI_C1q+ TLF+ Macrophages.csv")
rownames(C1q_MI_DEGenes) <- C1q_MI_DEGenes$gene
```

### Figure 1G: Volcano Plot of DEGenes in C1q Macs
```{r}
keyvals <- ifelse(
     C1q_MI_DEGenes$avg_log2FC > 0.3, 'firebrick4',
      ifelse(C1q_MI_DEGenes$avg_log2FC < -0.3, 'dodgerblue4',
        'gray50'))
  keyvals[is.na(keyvals)] <- 'gray50'
  names(keyvals)[keyvals == 'dodgerblue4'] <- "WT"
  names(keyvals)[keyvals == 'gray50'] <- 'No Significance'
  names(keyvals)[keyvals == 'firebrick4'] <- "MerKO"

lab_italics <- paste0("bolditalic('", rownames(C1q_MI_DEGenes), "')")

selectLab_italics = paste0("bolditalic('",
    c('Mrc1', 'Hpgd', 'Igf1', 'Igfbp4', 'mt-Nd3', 'Hpgds', 'Gas6', 'C1qc', 'Nr4a1', 'Lpl', 'Cd36', 'Axl', 'C1qa', 'Saa3', 'Fn1', 'S100a8', 'Marco', 'Arg1', 'S100a9', 'Plac8', 'Acod1', 'Retnlg', 'Slc16a3', 'Pgk1', 'Ltb4r1', 'Hif1a', 'Mmp9', 'Pkm', 'Il1b', 'Gapdh', 'Ccr2', 'Vegfa', 'Ifitm2', 'Ldha', 'Slc2a1', 'Ptgs2', 'Spp1', 'Rps28', 'Ifitm3', 'Ifitm1', 'mt-Cyb1', 'Fcrls', 'Prdx5', 'Lgmn', 'Pfn1', 'Hp'), "')")

Volcano_1 <- EnhancedVolcano(C1q_MI_DEGenes,
    title = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s))),
    titleLabSize = 40,
    subtitle = element_blank(),
    subtitleLabSize = 5,
    lab = lab_italics,
    selectLab = selectLab_italics,
    x = 'avg_log2FC',
    y = 'p_val_adj',
    xlab = bquote(~Log[2]~ 'Fold Change'),
    FCcutoff = 0.3, 
    pCutoff = NA,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 4.0,
    labSize = 8.0,
    labFace = 'bold',
    boxedLabels = T,
    parseLabels = T,
    colAlpha = 0.5,
    col = c("grey50", "firebrick3"),
    legendPosition = 'right',
    legendLabSize = 20,
    gridlines.major = F,
    gridlines.minor = F,
    colCustom = keyvals,
    drawConnectors = T,
    widthConnectors = 1.0,
    colConnectors = "black",
    max.overlaps = Inf, 
    arrowheads = F, 
    raster = T) + guides(color = guide_legend(override.aes = list(size = 12), ncol = 1, byrow = T)) + 
  scale_x_continuous(breaks=seq(-2,2,by=1)) +
  scale_color_manual(labels = c(expression(bolditalic("MerTK"^"-/-")),
                                expression(bolditalic("MerTK"^"+/+")), 
                                expression(bold("No Significance"))), values =  c( "firebrick4", "gray25", "gray90")) +
  theme(legend.text.align = 0, 
        legend.spacing.y = unit(0.8, "cm"), 
        plot.title = element_text(hjust = 0.5), 
        axis.title.x = element_text(face = "italic", family = "Helvetica", size = rel(3)), 
        axis.title.y = element_text(face = "italic", family = "Helvetica", size = rel(3)),
        axis.text.x = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        axis.text.y = element_text(face = "bold", family = "Helvetica", color = "black", size = rel(2)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.3)))
Volcano_1
```

```{r eval=FALSE, include=FALSE}
ggsave(filename = "MerKO_C1q_MI_DEGenes_Volcano.tiff", plot = Volcano_1, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/MerKO/", device = "tiff", dpi = 600, width = 20, height = 10, units = "in", compression = "lzw")
```

#### Regenerative and Reparative Genes
```{r}
Idents(Myeloid_Cells_WT) <- "Seurat_IDs"
DefaultAssay(Myeloid_Cells_WT) <- "RNA"

features.vln <- c("Igf2bp3", "Vegfb", "Tgfb1", "Timd4", "Il10", "Trem2", "Arg1", "Alox5ap")

VlnPlot <- VlnPlot(Myeloid_Cells_WT, features = features.vln,  split.by = "sample", group.by = "Seurat_IDs", cols = c("gray70", "deepskyblue2", "tomato", "firebrick3"), idents = c("C1q+ TLF+ Macrophages"), pt.size = 0.4, combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + 
    theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(3)), 
          axis.title.x = element_blank(), 
          axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.7)),
          axis.text.x =element_blank(), 
          axis.text.y = element_blank(), axis.ticks.y = element_blank(),
          axis.line = element_line(linewidth = rel(2))) + 
    NoLegend() 
}
VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, ncol = 4)
VlnPlot
```


### Venn Diagram
```{r}
C1q_MI_DEGenes <- read_csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/Sham Comparison/DEGs_WTShvMERSh_C1q+ TLF+ Macrophages.csv")
C1q_MI_DEGene_list <- C1q_MI_DEGenes$gene

Arg1_MI_DEGenes <- read_csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/Sham Comparison/DEGs_WTShvMERSh_Arg1+ Macrophages.csv")
Arg1_MI_DEGenes_list <- Arg1_MI_DEGenes$gene

Ccr2_MI_DEGenes <- read_csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/Sham Comparison/DEGs_WTShvMERSh_Ccr2+ Macrophages.csv")
Ccr2_MI_DEGenes_list <- Ccr2_MI_DEGenes$gene

Spp1_MI_DEGenes <- read_csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/Sham Comparison/DEGs_WTShvMERSh_Spp1+ Macrophages.csv")
Spp1_MI_DEGenes_list <- Spp1_MI_DEGenes$gene
```

```{r}
Sham_genes <- list("C1q+" = C1q_MI_DEGene_list, "Arg1+" = Arg1_MI_DEGenes_list, "Ccr2+" = Ccr2_MI_DEGenes_list, "Spp1+" = Spp1_MI_DEGenes_list)
```

```{r}
ggvenn(Sham_genes, fill_color = c("dodgerblue4", "lightcoral",  "firebrick3", "forestgreen"))
```
```{r}
v.table <- venn(Sham_genes)
print(v.table)

write.csv(attr(v.table,"intersections")$`C1q+:Arg1+:Ccr2+`, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/vtable.csv")
```


```{r}
C1q_MI_DEGenes <- read_csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/DEGs_WTMIvMERMI_C1q+ TLF+ Macrophages.csv")
C1q_MI_DEGene_list <- C1q_MI_DEGenes$gene

Arg1_MI_DEGenes <- read_csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/DEGs_WTMIvMERMI_Arg1+ Macrophages.csv")
Arg1_MI_DEGenes_list <- Arg1_MI_DEGenes$gene

Ccr2_MI_DEGenes <- read_csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/DEGs_WTMIvMERMI_Ccr2+ Macrophages.csv")
Ccr2_MI_DEGenes_list <- Ccr2_MI_DEGenes$gene

Spp1_MI_DEGenes <- read_csv("/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/DEGs_WTMIvMERMI_Spp1+ Macrophages.csv")
Spp1_MI_DEGenes_list <- Spp1_MI_DEGenes$gene
```

```{r}
MI_genes <- list("C1q+" = C1q_MI_DEGene_list, "Arg1+" = Arg1_MI_DEGenes_list, "Ccr2+" = Ccr2_MI_DEGenes_list, "Spp1+" = Spp1_MI_DEGenes_list)
```

```{r}
MI_Venn <- ggvenn(MI_genes, fill_color = c("dodgerblue4", "lightcoral",  "firebrick3", "forestgreen"))

MI_Venn 
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "MerKO_VennDiagram_MI.tiff", plot = MI_Venn, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/MerKO/", device = "tiff", dpi = 600, width = 10, height = 10, units = "in", compression = "lzw")
```

```{r}
v.table <- venn(MI_genes)
print(v.table)
write.csv(attr(v.table,"intersections")$`C1q+:Arg1+:Ccr2+`, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/vtable.csv")

view(attr(v.table,"intersections")$`C1q+:Arg1+:Ccr2+`)
```


```{r}
#C1q+ Specific
rownames(C1q_MI_DEGenes) <- C1q_MI_DEGenes$gene
C1q_specific <- attr(v.table,"intersections")$`C1q+:Arg1+:Spp1+`
C1q_specific_df <- C1q_MI_DEGenes[rownames(C1q_MI_DEGenes) %in% C1q_specific, ]
write.csv(C1q_specific_df, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/vtable_C1q_only.csv")

#C1q+ Specific
rownames(Arg1_MI_DEGenes) <- Arg1_MI_DEGenes$gene
Arg1_specific <- attr(v.table,"intersections")$`Arg1+:Spp1+`
Arg1_specific_df <- Arg1_MI_DEGenes[rownames(Arg1_MI_DEGenes) %in% Arg1_specific, ]
write.csv(Arg1_specific_df, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MI/vtable_Arg1_only.csv")
```

### Figure 5C: FeaturePlot of Mac Markers
```{r}
DefaultAssay(Myeloid_Cells_MerKO) <- "SCT"

features.markers <- c("Tbxas1", "Ptges", "Hpgds", "Prxl2b")

FeaturePlot_Cell_Markers <- FeaturePlot(Myeloid_Cells_MerKO,  features = features.markers, order = T, cols = c("gray89", "firebrick4"), min.cutoff = ("q50"), max.cutoff = ("q95"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] +
  theme(plot.title = element_text(hjust = 0.5, face = "bold.italic", family = "Helvetica", size = rel(2.25)),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) + NoLegend()
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 4)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "MerKO_Prostaglandins.tiff", plot = FeaturePlot_Cell_Markers, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/MerKO/", device = "tiff", dpi = 600, width = 16, height = 4, units = "in", compression = "lzw")
```

### Figure 5E: FeaturePlot of Mac Markers
```{r}
DefaultAssay(Myeloid_Cells_MerKO) <- "SCT"

features.markers <- c("Gpx4", "Alox5", "Alox5ap", "Alox15")

FeaturePlot_Cell_Markers <- FeaturePlot(Myeloid_Cells_MerKO,  features = features.markers, order = T, cols = c("gray89", "firebrick4"), min.cutoff = ("q40"), max.cutoff = ("q95"), combine = F)

for(i in 1:length(FeaturePlot_Cell_Markers)) {
  FeaturePlot_Cell_Markers[[i]] <- FeaturePlot_Cell_Markers[[i]] +
  theme(plot.title = element_text(hjust = 0.5, face = "bold.italic", family = "Helvetica", size = rel(2.25)),
        axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(), axis.line = element_blank()) + NoLegend()
  
}

FeaturePlot_Cell_Markers <- cowplot::plot_grid(plotlist = FeaturePlot_Cell_Markers, ncol = 4)
FeaturePlot_Cell_Markers

FeaturePlot_Cell_Markers <- plot_grid(NULL, FeaturePlot_Cell_Markers, NULL, NULL,
                                        nrow = 2,
                                        rel_widths = c(0.05, 0.95),
                                        rel_heights = c(0.94, 0.06))
```

```{r eval=FALSE, include=FALSE}
ggsave(filename = "MerKO_HETE_Enzymes.tiff", plot = FeaturePlot_Cell_Markers, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/MerKO/", device = "tiff", dpi = 600, width = 16, height = 4, units = "in", compression = "lzw")
```

### Figure 5F: Violin Plot of Arachidonic Acid Enzymes in MerKO and Adult Macs
```{r}
Myeloid_Cells_All_Conditions <- merge(Myeloid_Cells_WT, Myeloid_Cells_MerKO, merge.dr = c("umap"))

levels <- c("C1q+ TLF+ Macrophages", "Prolif. C1q+ Macrophages", "Arg1+ Macrophages", "Ccr2+ Macrophages", "Spp1+ Macrophages", "Isg Cells", "Ly6chi Monocytes", "Ly6clo Monocytes", "cDC2s", "cDC1s", "pDCs", "Ccr7+ DCs", "Dying Cells")
Myeloid_Cells_All_Conditions$Seurat_IDs <- factor(x = Myeloid_Cells_All_Conditions$Seurat_IDs, levels = levels)

levels <- c("WT_P1_Sh", "WT_P1_MI", "MER_Sh", "MER_MI", "WT_Ad_Sh", "WT_Ad_MI")
Myeloid_Cells_All_Conditions$sample <- factor(x = Myeloid_Cells_All_Conditions$sample, levels = levels)

cols.use <- c("dodgerblue4", "deepskyblue1", "lightcoral",  "firebrick1", "forestgreen", "lightpink", "firebrick4", "tan4", "darkorange", "darkorange3", "lightgoldenrod2", "salmon1", "gray")


Idents(Myeloid_Cells_All_Conditions) <- "Seurat_IDs"
Dimplot_Myeloid_labels <- DimPlot(Myeloid_Cells_All_Conditions, reduction = "umap", label = F, cols = cols.use) + guides(color = guide_legend(override.aes = list(size=4), ncol=1)) +
  geom_segment(aes(x=-11, y=-11, xend=-5, yend=-11), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  geom_segment(aes(x=-11, y=-11, xend=-11, yend=-5), arrow = arrow(length=unit(0.3, 'cm'), type = "closed")) +
  labs(x = "UMAP-1", y = "UMAP-2") +
  theme_void() +
  theme(axis.title.x = element_text(hjust = 0.09, vjust = 3.25, face = "bold", family = "Helvetica", size = rel(1.5)),
        axis.title.y = element_text(hjust = 0.10, vjust = -3.35, angle=90, face = "bold", family = "Helvetica", size = rel(1.5)),
        legend.text = element_text(face = "bold", family = "Helvetica", size = rel(1.2)),
        legend.text.align = 0) + 
  scale_color_manual(labels = c(expression(bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bold(Prolif.)~bolditalic("C1q"^"+"~TLF^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Arg1"^"+")~bold(M*Phi*s)), 
                                expression(bolditalic("Ccr2"^"+"~TLF^"-")~bold(M*Phi*s)),
                                expression(bolditalic("Spp1"^"+")~bold(M*Phi*s)),
                                expression(bolditalic("Isg"^"+")~bold(Cells)), 
                                expression(bolditalic("Ly6c"^"HI")~bold(Monocytes)), 
                                expression(bolditalic("Ly6c"^"LO")~bold(Monocytes)), 
                                expression(bold("cDC2s")), 
                                expression(bold("cDC1s")), 
                                expression(bold("pDCs")), 
                                expression(bold("Ccr7+ DCs")), 
                                expression(bold("Dying Cells"))), 
                     values = cols.use)

Dimplot_Myeloid_labels
```
```{r}
Idents(Myeloid_Cells_All_Conditions) <- "Seurat_IDs"
DefaultAssay(Myeloid_Cells_All_Conditions) <- "RNA"

features.vln <- c("mt-Co1", "Tbxas1", "Hpgds", "Ptgs2", "Prxl2b", "Alox5ap")

VlnPlot <- VlnPlot(Myeloid_Cells_All_Conditions, features = features.vln, split.by = "sample", group.by = "Seurat_IDs", idents = c("Ccr2+ Macrophages"), pt.size = 0.01, cols = c("gray75", "gray25","firebrick1", "firebrick4", "orange", "darkorange3"), combine = F)
for(i in 1:length(VlnPlot)) {
  VlnPlot[[i]] <- VlnPlot[[i]] + labs(title = features.vln[[i]], y = "Expression") + theme(plot.title = element_text(face = "bold.italic", family = "Helvetica", size = rel(1.5)), axis.title.x = element_blank(), axis.title.y = element_text(face = "bold", family = "Helvetica", size = rel(1.5)), axis.text.x = element_blank(), axis.text.y = element_text(face = "bold", family = "Helvetica", size = rel(1.5))) + NoLegend()
}

Regenerative_VlnPlot <- cowplot::plot_grid(plotlist = VlnPlot, nrow = 2)
Regenerative_VlnPlot
```
```{r eval=FALSE, include=FALSE}
ggsave(filename = "MerKO_VlnPlot_AA_Enzymes_Ccr2.tiff", plot = Regenerative_VlnPlot, path = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/FIGURES/MerKO/", device = "tiff", dpi = 600, width = 8, height = 6, units = "in", compression = "lzw")
```



```{r}
Idents(Myeloid_Cells_MerKO) <- "cluster.sample"

MerKO_Average_Expression <- AverageExpression(Myeloid_Cells_MerKO, assays = "RNA", group.by = "cluster_sample")

write.csv(MerKO_Average_Expression, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MerKO_Average_Expression.csv")
```

```{r}
Idents(Myeloid_Cells_MerKO) <- "cluster_sample"
MerKO_Average_Expression_FC <- FoldChange(Myeloid_Cells_MerKO, ident.1 = "C1q+ TLF+ Macrophages_WT_P1_MI", ident.2 = "C1q+ TLF+ Macrophages_MER_MI")

write.csv(MerKO_Average_Expression_FC, file = "/projects/p31275/Lantz/R/Lantz_etal_Manuscript/RESULTS/MerKO/MerKO_Average_Expression_FC.csv")
```














